From dfd1424450a3bf1228b84ed32317f8949e32e603 Mon Sep 17 00:00:00 2001
From: Alexandre Julliard <julliard@winehq.org>
Date: Thu, 5 Jun 2025 12:00:59 +0200
Subject: [PATCH 384/406] ntdll: Add explicit ids to a number of syscalls.

For the ones that seem to be mostly stable across Windows versions,
based on the syscall table by Mateusz "j00ru" Jurczyk.
---
 dlls/ntdll/ntdll.spec   | 172 +++----
 dlls/ntdll/ntsyscalls.h | 984 ++++++++++++++++++++--------------------
 2 files changed, 578 insertions(+), 578 deletions(-)

diff --git a/dlls/ntdll/ntdll.spec b/dlls/ntdll/ntdll.spec
index 3fd05949d00..10445c797c3 100644
--- a/dlls/ntdll/ntdll.spec
+++ b/dlls/ntdll/ntdll.spec
@@ -130,18 +130,18 @@
 @ extern NlsAnsiCodePage
 @ extern NlsMbCodePageTag
 @ extern NlsMbOemCodePageTag
-@ stdcall -syscall NtAcceptConnectPort(ptr long ptr long ptr ptr)
-@ stdcall -syscall NtAccessCheck(ptr long long ptr ptr ptr ptr ptr)
-@ stdcall -syscall NtAccessCheckAndAuditAlarm(ptr long ptr ptr ptr long ptr long ptr ptr ptr)
+@ stdcall -syscall=0x0002 NtAcceptConnectPort(ptr long ptr long ptr ptr)
+@ stdcall -syscall=0x0000 NtAccessCheck(ptr long long ptr ptr ptr ptr ptr)
+@ stdcall -syscall=0x0029 NtAccessCheckAndAuditAlarm(ptr long ptr ptr ptr long ptr long ptr ptr ptr)
 # @ stub NtAccessCheckByType
 # @ stub NtAccessCheckByTypeAndAuditAlarm
 # @ stub NtAccessCheckByTypeResultList
 # @ stub NtAccessCheckByTypeResultListAndAuditAlarm
 # @ stub NtAccessCheckByTypeResultListAndAuditAlarmByHandle
-@ stdcall -syscall NtAddAtom(ptr long ptr)
+@ stdcall -syscall=0x0047 NtAddAtom(ptr long ptr)
 # @ stub NtAddBootEntry
 @ stdcall -syscall NtAdjustGroupsToken(long long ptr long ptr ptr)
-@ stdcall -syscall NtAdjustPrivilegesToken(long long ptr long ptr ptr)
+@ stdcall -syscall=0x0041 NtAdjustPrivilegesToken(long long ptr long ptr ptr)
 @ stdcall -syscall NtAlertMultipleThreadByThreadId(ptr long ptr ptr)
 @ stdcall -syscall NtAlertResumeThread(long ptr)
 @ stdcall -syscall NtAlertThread(long)
@@ -150,18 +150,18 @@
 @ stdcall -syscall NtAllocateReserveObject(ptr ptr long)
 # @ stub NtAllocateUserPhysicalPages
 @ stdcall -syscall NtAllocateUuids(ptr ptr ptr ptr)
-@ stdcall -syscall NtAllocateVirtualMemory(long ptr long ptr long long)
+@ stdcall -syscall=0x0018 NtAllocateVirtualMemory(long ptr long ptr long long)
 @ stdcall -syscall NtAllocateVirtualMemoryEx(long ptr ptr long long ptr long)
 @ stdcall -syscall NtAreMappedFilesTheSame(ptr ptr)
 @ stdcall -syscall NtAssignProcessToJobObject(long long)
-@ stdcall -syscall NtCallbackReturn(ptr long long)
+@ stdcall -syscall=0x0005 NtCallbackReturn(ptr long long)
 # @ stub NtCancelDeviceWakeupRequest
-@ stdcall -syscall NtCancelIoFile(long ptr)
+@ stdcall -syscall=0x005d NtCancelIoFile(long ptr)
 @ stdcall -syscall NtCancelIoFileEx(long ptr ptr)
 @ stdcall -syscall NtCancelSynchronousIoFile(long ptr ptr)
-@ stdcall -syscall NtCancelTimer(long ptr)
-@ stdcall -syscall NtClearEvent(long)
-@ stdcall -syscall NtClose(long)
+@ stdcall -syscall=0x0061 NtCancelTimer(long ptr)
+@ stdcall -syscall=0x003e NtClearEvent(long)
+@ stdcall -syscall=0x000f NtClose(long)
 # @ stub NtCloseObjectAuditAlarm
 @ stdcall -syscall NtCommitTransaction(long long)
 # @ stub NtCompactKeys
@@ -170,18 +170,18 @@
 @ stdcall -syscall NtCompleteConnectPort(ptr)
 # @ stub NtCompressKey
 @ stdcall -syscall NtConnectPort(ptr ptr ptr ptr ptr ptr ptr ptr)
-@ stdcall -syscall NtContinue(ptr long)
+@ stdcall -syscall=0x0043 NtContinue(ptr long)
 @ stdcall -syscall NtContinueEx(ptr ptr)
 @ stdcall -syscall NtConvertBetweenAuxiliaryCounterAndPerformanceCounter(long ptr ptr ptr)
-@ stdcall -syscall NtCreateDebugObject(ptr long ptr long)
+@ stdcall -syscall=0x00a6 NtCreateDebugObject(ptr long ptr long)
 @ stdcall -syscall NtCreateDirectoryObject(ptr long ptr)
-@ stdcall -syscall NtCreateEvent(ptr long ptr long long)
+@ stdcall -syscall=0x0048 NtCreateEvent(ptr long ptr long long)
 # @ stub NtCreateEventPair
-@ stdcall -syscall NtCreateFile(ptr long ptr ptr ptr long long long long ptr long)
+@ stdcall -syscall=0x0055 NtCreateFile(ptr long ptr ptr ptr long long long long ptr long)
 @ stdcall -syscall NtCreateIoCompletion(ptr long ptr long)
 @ stdcall -syscall NtCreateJobObject(ptr long ptr)
 # @ stub NtCreateJobSet
-@ stdcall -syscall NtCreateKey(ptr long ptr long ptr long ptr)
+@ stdcall -syscall=0x001d NtCreateKey(ptr long ptr long ptr long ptr)
 @ stdcall -syscall NtCreateKeyTransacted(ptr long ptr long ptr long long ptr)
 @ stdcall -syscall NtCreateKeyedEvent(ptr long ptr long)
 @ stdcall -syscall NtCreateLowBoxToken(ptr long long ptr ptr long ptr long ptr)
@@ -193,11 +193,11 @@
 # @ stub NtCreateProcess
 # @ stub NtCreateProcessEx
 # @ stub NtCreateProfile
-@ stdcall -syscall NtCreateSection(ptr long ptr ptr long long long)
+@ stdcall -syscall=0x004a NtCreateSection(ptr long ptr ptr long long long)
 @ stdcall -syscall NtCreateSectionEx(ptr long ptr ptr long long long ptr long)
 @ stdcall -syscall NtCreateSemaphore(ptr long ptr long long)
 @ stdcall -syscall NtCreateSymbolicLinkObject(ptr long ptr ptr)
-@ stdcall -syscall NtCreateThread(ptr long ptr long ptr ptr ptr long)
+@ stdcall -syscall=0x004e NtCreateThread(ptr long ptr long ptr ptr ptr long)
 @ stdcall -syscall NtCreateThreadEx(ptr long ptr long ptr ptr long long long long ptr)
 @ stdcall -syscall NtCreateTimer(ptr long ptr long)
 @ stdcall -syscall NtCreateToken(ptr long ptr long ptr ptr ptr ptr ptr ptr ptr ptr ptr)
@@ -207,26 +207,26 @@
 @ stdcall -arch=i386 NtCurrentTeb()
 @ stdcall -syscall NtDebugActiveProcess(long long)
 @ stdcall -syscall NtDebugContinue(long ptr long)
-@ stdcall -syscall NtDelayExecution(long ptr)
+@ stdcall -syscall=0x0034 NtDelayExecution(long ptr)
 @ stdcall -syscall NtDeleteAtom(long)
 # @ stub NtDeleteBootEntry
 @ stdcall -syscall NtDeleteFile(ptr)
 @ stdcall -syscall NtDeleteKey(long)
 # @ stub NtDeleteObjectAuditAlarm
 @ stdcall -syscall NtDeleteValueKey(long ptr)
-@ stdcall -syscall NtDeviceIoControlFile(long long ptr ptr ptr long ptr long ptr long)
+@ stdcall -syscall=0x0007 NtDeviceIoControlFile(long long ptr ptr ptr long ptr long ptr long)
 @ stdcall -syscall NtDisplayString(ptr)
-@ stdcall -syscall NtDuplicateObject(long long long ptr long long long)
-@ stdcall -syscall NtDuplicateToken(long long ptr long long ptr)
+@ stdcall -syscall=0x003c NtDuplicateObject(long long long ptr long long long)
+@ stdcall -syscall=0x0042 NtDuplicateToken(long long ptr long long ptr)
 # @ stub NtEnumerateBootEntries
 # @ stub NtEnumerateBus
-@ stdcall -syscall NtEnumerateKey(long long long ptr long ptr)
+@ stdcall -syscall=0x0032 NtEnumerateKey(long long long ptr long ptr)
 # @ stub NtEnumerateSystemEnvironmentValuesEx
-@ stdcall -syscall NtEnumerateValueKey(long long long ptr long ptr)
+@ stdcall -syscall=0x0013 NtEnumerateValueKey(long long long ptr long ptr)
 # @ stub NtExtendSection
 @ stdcall -syscall NtFilterToken(long long ptr ptr ptr ptr)
-@ stdcall -syscall NtFindAtom(ptr long ptr)
-@ stdcall -syscall NtFlushBuffersFile(long ptr)
+@ stdcall -syscall=0x0014 NtFindAtom(ptr long ptr)
+@ stdcall -syscall=0x004b NtFlushBuffersFile(long ptr)
 @ stdcall -syscall NtFlushBuffersFileEx(long long ptr long ptr)
 @ stdcall -syscall NtFlushInstructionCache(long ptr long)
 @ stdcall -syscall NtFlushKey(long)
@@ -234,8 +234,8 @@
 @ stdcall -syscall NtFlushVirtualMemory(long ptr ptr long)
 # @ stub NtFlushWriteBuffer
 # @ stub NtFreeUserPhysicalPages
-@ stdcall -syscall NtFreeVirtualMemory(long ptr ptr long)
-@ stdcall -syscall NtFsControlFile(long long ptr ptr ptr long ptr long ptr long)
+@ stdcall -syscall=0x001e NtFreeVirtualMemory(long ptr ptr long)
+@ stdcall -syscall=0x0039 NtFsControlFile(long long ptr ptr ptr long ptr long ptr long)
 @ stdcall -norelay -syscall NtGetContextThread(long ptr)
 @ stdcall -syscall NtGetCurrentProcessorNumber()
 # @ stub NtGetDevicePowerState
@@ -251,7 +251,7 @@
 @ stdcall -syscall NtInitializeNlsFiles(ptr ptr ptr)
 # @ stub NtInitializeRegistry
 @ stdcall -syscall NtInitiatePowerAction (long long long long)
-@ stdcall -syscall NtIsProcessInJob(long long)
+@ stdcall -syscall=0x004f NtIsProcessInJob(long long)
 # @ stub NtIsSystemResumeAutomatic
 @ stdcall -syscall NtListenPort(ptr ptr)
 @ stdcall -syscall NtLoadDriver(ptr)
@@ -266,118 +266,118 @@
 @ stdcall -syscall NtMakeTemporaryObject(long)
 # @ stub NtMapUserPhysicalPages
 # @ stub NtMapUserPhysicalPagesScatter
-@ stdcall -syscall NtMapViewOfSection(long long ptr long long ptr ptr long long long)
+@ stdcall -syscall=0x0028 NtMapViewOfSection(long long ptr long long ptr ptr long long long)
 @ stdcall -syscall NtMapViewOfSectionEx(long long ptr ptr ptr long long ptr long)
 # @ stub NtModifyBootEntry
 @ stdcall -syscall NtNotifyChangeDirectoryFile(long long ptr ptr ptr ptr long long long)
 @ stdcall -syscall NtNotifyChangeKey(long long ptr ptr ptr long long ptr long long)
 @ stdcall -syscall NtNotifyChangeMultipleKeys(long long ptr long ptr ptr ptr long long ptr long long)
-@ stdcall -syscall NtOpenDirectoryObject(ptr long ptr)
-@ stdcall -syscall NtOpenEvent(ptr long ptr)
+@ stdcall -syscall=0x0058 NtOpenDirectoryObject(ptr long ptr)
+@ stdcall -syscall=0x0040 NtOpenEvent(ptr long ptr)
 # @ stub NtOpenEventPair
-@ stdcall -syscall NtOpenFile(ptr long ptr ptr long long)
+@ stdcall -syscall=0x0033 NtOpenFile(ptr long ptr ptr long long)
 @ stdcall -syscall NtOpenIoCompletion(ptr long ptr)
 @ stdcall -syscall NtOpenJobObject(ptr long ptr)
-@ stdcall -syscall NtOpenKey(ptr long ptr)
+@ stdcall -syscall=0x0012 NtOpenKey(ptr long ptr)
 @ stdcall -syscall NtOpenKeyEx(ptr long ptr long)
 @ stdcall -syscall NtOpenKeyTransacted(ptr long ptr long)
 @ stdcall -syscall NtOpenKeyTransactedEx(ptr long ptr long long)
 @ stdcall -syscall NtOpenKeyedEvent(ptr long ptr)
 @ stdcall -syscall NtOpenMutant(ptr long ptr)
 # @ stub NtOpenObjectAuditAlarm
-@ stdcall -syscall NtOpenProcess(ptr long ptr ptr)
+@ stdcall -syscall=0x0026 NtOpenProcess(ptr long ptr ptr)
 @ stdcall -syscall NtOpenProcessToken(long long ptr)
-@ stdcall -syscall NtOpenProcessTokenEx(long long long ptr)
-@ stdcall -syscall NtOpenSection(ptr long ptr)
+@ stdcall -syscall=0x0030 NtOpenProcessTokenEx(long long long ptr)
+@ stdcall -syscall=0x0037 NtOpenSection(ptr long ptr)
 @ stdcall -syscall NtOpenSemaphore(ptr long ptr)
 @ stdcall -syscall NtOpenSymbolicLinkObject (ptr long ptr)
 @ stdcall -syscall NtOpenThread(ptr long ptr ptr)
-@ stdcall -syscall NtOpenThreadToken(long long long ptr)
-@ stdcall -syscall NtOpenThreadTokenEx(long long long long ptr)
+@ stdcall -syscall=0x0024 NtOpenThreadToken(long long long ptr)
+@ stdcall -syscall=0x002f NtOpenThreadTokenEx(long long long long ptr)
 @ stdcall -syscall NtOpenTimer(ptr long ptr)
 # @ stub NtPlugPlayControl
-@ stdcall -syscall NtPowerInformation(long ptr long ptr long)
+@ stdcall -syscall=0x005f NtPowerInformation(long ptr long ptr long)
 @ stdcall -syscall NtPrivilegeCheck(ptr ptr ptr)
 # @ stub NtPrivilegeObjectAuditAlarm
 # @ stub NtPrivilegedServiceAuditAlarm
-@ stdcall -syscall NtProtectVirtualMemory(long ptr ptr long ptr)
+@ stdcall -syscall=0x0050 NtProtectVirtualMemory(long ptr ptr long ptr)
 @ stdcall -syscall NtPulseEvent(long ptr)
-@ stdcall -syscall NtQueryAttributesFile(ptr ptr)
+@ stdcall -syscall=0x003d NtQueryAttributesFile(ptr ptr)
 # @ stub NtQueryBootEntryOrder
 # @ stub NtQueryBootOptions
 # @ stub NtQueryDebugFilterState
-@ stdcall -syscall NtQueryDefaultLocale(long ptr)
-@ stdcall -syscall NtQueryDefaultUILanguage(ptr)
-@ stdcall -syscall NtQueryDirectoryFile(long long ptr ptr ptr ptr long long long ptr long)
+@ stdcall -syscall=0x0015 NtQueryDefaultLocale(long ptr)
+@ stdcall -syscall=0x0044 NtQueryDefaultUILanguage(ptr)
+@ stdcall -syscall=0x0035 NtQueryDirectoryFile(long long ptr ptr ptr ptr long long long ptr long)
 @ stdcall -syscall NtQueryDirectoryObject(long ptr long long long ptr ptr)
 @ stdcall -syscall NtQueryEaFile(long ptr ptr long long ptr long ptr long)
-@ stdcall -syscall NtQueryEvent(long long ptr long ptr)
+@ stdcall -syscall=0x0056 NtQueryEvent(long long ptr long ptr)
 @ stdcall -syscall NtQueryFullAttributesFile(ptr ptr)
 @ stdcall -syscall NtQueryInformationAtom(long long ptr long ptr)
-@ stdcall -syscall NtQueryInformationFile(long ptr ptr long long)
+@ stdcall -syscall=0x0011 NtQueryInformationFile(long ptr ptr long long)
 @ stdcall -syscall NtQueryInformationJobObject(long long ptr long ptr)
 # @ stub NtQueryInformationPort
-@ stdcall -syscall NtQueryInformationProcess(long long ptr long ptr)
-@ stdcall -syscall NtQueryInformationThread(long long ptr long ptr)
-@ stdcall -syscall NtQueryInformationToken(long long ptr long ptr)
+@ stdcall -syscall=0x0019 NtQueryInformationProcess(long long ptr long ptr)
+@ stdcall -syscall=0x0025 NtQueryInformationThread(long long ptr long ptr)
+@ stdcall -syscall=0x0021 NtQueryInformationToken(long long ptr long ptr)
 @ stdcall -syscall NtQueryInstallUILanguage(ptr)
 # @ stub NtQueryIntervalProfile
 @ stdcall -syscall NtQueryIoCompletion(long long ptr long ptr)
-@ stdcall -syscall NtQueryKey(long long ptr long ptr)
+@ stdcall -syscall=0x0016 NtQueryKey(long long ptr long ptr)
 @ stdcall -syscall NtQueryLicenseValue(ptr ptr ptr long ptr)
 @ stdcall -syscall NtQueryMultipleValueKey(long ptr long ptr long ptr)
 @ stdcall -syscall NtQueryMutant(long long ptr long ptr)
-@ stdcall -syscall NtQueryObject(long long ptr long ptr)
+@ stdcall -syscall=0x0010 NtQueryObject(long long ptr long ptr)
 # @ stub NtQueryOpenSubKeys
-@ stdcall -syscall NtQueryPerformanceCounter(ptr ptr)
+@ stdcall -syscall=0x0031 NtQueryPerformanceCounter(ptr ptr)
 # @ stub NtQueryPortInformationProcess
 # @ stub NtQueryQuotaInformationFile
-@ stdcall -syscall NtQuerySection(long long ptr long ptr)
+@ stdcall -syscall=0x0051 NtQuerySection(long long ptr long ptr)
 @ stdcall -syscall NtQuerySecurityObject(long long ptr long ptr)
 @ stdcall -syscall NtQuerySemaphore (long long ptr long ptr)
 @ stdcall -syscall NtQuerySymbolicLinkObject(long ptr ptr)
 @ stdcall -syscall NtQuerySystemEnvironmentValue(ptr ptr long ptr)
 @ stdcall -syscall NtQuerySystemEnvironmentValueEx(ptr ptr ptr ptr ptr)
-@ stdcall -syscall NtQuerySystemInformation(long ptr long ptr)
+@ stdcall -syscall=0x0036 NtQuerySystemInformation(long ptr long ptr)
 @ stdcall -syscall NtQuerySystemInformationEx(long ptr long ptr long ptr)
-@ stdcall -syscall NtQuerySystemTime(ptr)
-@ stdcall -syscall NtQueryTimer(ptr long ptr long ptr)
+@ stdcall -syscall=0x005a NtQuerySystemTime(ptr)
+@ stdcall -syscall=0x0038 NtQueryTimer(ptr long ptr long ptr)
 @ stdcall -syscall NtQueryTimerResolution(ptr ptr ptr)
-@ stdcall -syscall NtQueryValueKey(long ptr long ptr long ptr)
-@ stdcall -syscall NtQueryVirtualMemory(long ptr long ptr long ptr)
-@ stdcall -syscall NtQueryVolumeInformationFile(long ptr ptr long long)
-@ stdcall -syscall NtQueueApcThread(long ptr long long long)
+@ stdcall -syscall=0x0017 NtQueryValueKey(long ptr long ptr long ptr)
+@ stdcall -syscall=0x0023 NtQueryVirtualMemory(long ptr long ptr long ptr)
+@ stdcall -syscall=0x0049 NtQueryVolumeInformationFile(long ptr ptr long long)
+@ stdcall -syscall=0x0045 NtQueueApcThread(long ptr long long long)
 @ stdcall -syscall NtQueueApcThreadEx(long long ptr long long long)
 @ stdcall -syscall NtRaiseException(ptr ptr long)
 @ stdcall -syscall NtRaiseHardError(long long long ptr long ptr)
-@ stdcall -syscall NtReadFile(long long ptr ptr ptr ptr long ptr ptr)
-@ stdcall -syscall NtReadFileScatter(long long ptr ptr ptr ptr long ptr ptr)
+@ stdcall -syscall=0x0006 NtReadFile(long long ptr ptr ptr ptr long ptr ptr)
+@ stdcall -syscall=0x002e NtReadFileScatter(long long ptr ptr ptr ptr long ptr ptr)
 # @ stub NtReadRequestData
-@ stdcall -syscall NtReadVirtualMemory(long ptr ptr long ptr)
+@ stdcall -syscall=0x003f NtReadVirtualMemory(long ptr ptr long ptr)
 # @ stub NtRegisterNewDevice
 @ stdcall -syscall NtRegisterThreadTerminatePort(ptr)
 @ stdcall -syscall NtReleaseKeyedEvent(long ptr long ptr)
-@ stdcall -syscall NtReleaseMutant(long ptr)
+@ stdcall -syscall=0x0020 NtReleaseMutant(long ptr)
 # @ stub NtReleaseProcessMutant
-@ stdcall -syscall NtReleaseSemaphore(long long ptr)
-@ stdcall -syscall NtRemoveIoCompletion(ptr ptr ptr ptr ptr)
+@ stdcall -syscall=0x000a NtReleaseSemaphore(long long ptr)
+@ stdcall -syscall=0x0009 NtRemoveIoCompletion(ptr ptr ptr ptr ptr)
 @ stdcall -syscall NtRemoveIoCompletionEx(ptr ptr long ptr ptr long)
 @ stdcall -syscall NtRemoveProcessDebug(long long)
 @ stdcall -syscall NtRenameKey(long ptr)
 @ stdcall -syscall NtReplaceKey(ptr long ptr)
 # @ stub NtReplyPort
-@ stdcall -syscall NtReplyWaitReceivePort(ptr ptr ptr ptr)
+@ stdcall -syscall=0x000b NtReplyWaitReceivePort(ptr ptr ptr ptr)
 # @ stub NtReplyWaitReceivePortEx
 # @ stub NtReplyWaitReplyPort
 # @ stub NtRequestDeviceWakeup
 # @ stub NtRequestPort
-@ stdcall -syscall NtRequestWaitReplyPort(ptr ptr ptr)
+@ stdcall -syscall=0x0022 NtRequestWaitReplyPort(ptr ptr ptr)
 # @ stub NtRequestWakeupLatency
 @ stdcall -syscall NtResetEvent(long ptr)
 @ stdcall -syscall NtResetWriteWatch(long ptr long)
 @ stdcall -syscall NtRestoreKey(long long long)
 @ stdcall -syscall NtResumeProcess(long)
-@ stdcall -syscall NtResumeThread(long ptr)
+@ stdcall -syscall=0x0052 NtResumeThread(long ptr)
 @ stdcall -syscall NtRollbackTransaction(long long)
 @ stdcall -syscall NtSaveKey(long long)
 # @ stub NtSaveKeyEx
@@ -391,18 +391,18 @@
 @ stdcall -syscall NtSetDefaultLocale(long long)
 @ stdcall -syscall NtSetDefaultUILanguage(long)
 @ stdcall -syscall NtSetEaFile(long ptr ptr long)
-@ stdcall -syscall NtSetEvent(long ptr)
+@ stdcall -syscall=0x000e NtSetEvent(long ptr)
 # @ stub NtSetEventBoostPriority
 # @ stub NtSetHighEventPair
 # @ stub NtSetHighWaitLowEventPair
 # @ stub NtSetHighWaitLowThread
 @ stdcall -syscall NtSetInformationDebugObject(long long ptr long ptr)
-@ stdcall -syscall NtSetInformationFile(long ptr ptr long long)
+@ stdcall -syscall=0x0027 NtSetInformationFile(long ptr ptr long long)
 @ stdcall -syscall NtSetInformationJobObject(long long ptr long)
 @ stdcall -syscall NtSetInformationKey(long long ptr long)
-@ stdcall -syscall NtSetInformationObject(long long ptr long)
-@ stdcall -syscall NtSetInformationProcess(long long ptr long)
-@ stdcall -syscall NtSetInformationThread(long long ptr long)
+@ stdcall -syscall=0x005c NtSetInformationObject(long long ptr long)
+@ stdcall -syscall=0x001c NtSetInformationProcess(long long ptr long)
+@ stdcall -syscall=0x000d NtSetInformationThread(long long ptr long)
 @ stdcall -syscall NtSetInformationToken(long long ptr long)
 @ stdcall -syscall NtSetInformationVirtualMemory(long long ptr ptr ptr long)
 @ stdcall -syscall NtSetIntervalProfile(long long)
@@ -420,10 +420,10 @@
 # @ stub NtSetSystemPowerState
 @ stdcall -syscall NtSetSystemTime(ptr ptr)
 @ stdcall -syscall NtSetThreadExecutionState(long ptr)
-@ stdcall -syscall NtSetTimer(long ptr ptr ptr long long ptr)
+@ stdcall -syscall=0x0062 NtSetTimer(long ptr ptr ptr long long ptr)
 @ stdcall -syscall NtSetTimerResolution(long long ptr)
 # @ stub NtSetUuidSeed
-@ stdcall -syscall NtSetValueKey(long ptr long long ptr long)
+@ stdcall -syscall=0x0060 NtSetValueKey(long ptr long long ptr long)
 @ stdcall -syscall NtSetVolumeInformationFile(long ptr ptr long long)
 @ stdcall -syscall NtShutdownSystem(long)
 @ stdcall -syscall NtSignalAndWaitForSingleObject(long long long ptr)
@@ -433,8 +433,8 @@
 @ stdcall -syscall NtSuspendThread(long ptr)
 @ stdcall -syscall NtSystemDebugControl(long ptr long ptr long ptr)
 @ stdcall -syscall NtTerminateJobObject(long long)
-@ stdcall -syscall NtTerminateProcess(long long)
-@ stdcall -syscall NtTerminateThread(long long)
+@ stdcall -syscall=0x002c NtTerminateProcess(long long)
+@ stdcall -syscall=0x0053 NtTerminateThread(long long)
 @ stdcall -syscall NtTestAlert()
 @ stdcall -syscall NtTraceControl(long ptr long ptr long long)
 # @ stub NtTraceEvent
@@ -444,16 +444,16 @@
 # @ stub NtUnloadKeyEx
 @ stdcall -syscall NtUnlockFile(long ptr ptr ptr ptr)
 @ stdcall -syscall NtUnlockVirtualMemory(long ptr ptr long)
-@ stdcall -syscall NtUnmapViewOfSection(long ptr)
+@ stdcall -syscall=0x002a NtUnmapViewOfSection(long ptr)
 @ stdcall -syscall NtUnmapViewOfSectionEx(long ptr long)
 # @ stub NtVdmControl
 # @ stub NtW32Call
 @ stdcall -syscall NtWaitForAlertByThreadId(ptr ptr)
 @ stdcall -syscall NtWaitForDebugEvent(long long ptr ptr)
 @ stdcall -syscall NtWaitForKeyedEvent(long ptr long ptr)
-@ stdcall -syscall NtWaitForMultipleObjects(long ptr long long ptr)
+@ stdcall -syscall=0x005b NtWaitForMultipleObjects(long ptr long long ptr)
 # @ stub NtWaitForProcessMutant
-@ stdcall -syscall NtWaitForSingleObject(long long ptr)
+@ stdcall -syscall=0x0004 NtWaitForSingleObject(long long ptr)
 # @ stub NtWaitHighEventPair
 # @ stub NtWaitLowEventPair
 @ stdcall -syscall -arch=win32 NtWow64AllocateVirtualMemory64(long ptr int64 ptr long long)
@@ -462,11 +462,11 @@
 @ stdcall -syscall -arch=win32 NtWow64QueryInformationProcess64(long long ptr long ptr)
 @ stdcall -syscall -arch=win32 NtWow64ReadVirtualMemory64(long int64 ptr int64 ptr)
 @ stdcall -syscall -arch=win32 NtWow64WriteVirtualMemory64(long int64 ptr int64 ptr)
-@ stdcall -syscall NtWriteFile(long long ptr ptr ptr ptr long ptr ptr)
-@ stdcall -syscall NtWriteFileGather(long long ptr ptr ptr ptr long ptr ptr)
+@ stdcall -syscall=0x0008 NtWriteFile(long long ptr ptr ptr ptr long ptr ptr)
+@ stdcall -syscall=0x001b NtWriteFileGather(long long ptr ptr ptr ptr long ptr ptr)
 # @ stub NtWriteRequestData
-@ stdcall -syscall NtWriteVirtualMemory(long ptr ptr long ptr)
-@ stdcall -syscall NtYieldExecution()
+@ stdcall -syscall=0x003a NtWriteVirtualMemory(long ptr ptr long ptr)
+@ stdcall -syscall=0x0046 NtYieldExecution()
 @ stdcall NtdllDefWindowProc_A(long long long long)
 @ stdcall NtdllDefWindowProc_W(long long long long)
 @ stdcall NtdllDialogWndProc_A(long long long long)
diff --git a/dlls/ntdll/ntsyscalls.h b/dlls/ntdll/ntsyscalls.h
index 7623f4f9139..8860c483a80 100644
--- a/dlls/ntdll/ntsyscalls.h
+++ b/dlls/ntdll/ntsyscalls.h
@@ -1,502 +1,502 @@
 /* Automatically generated by tools/make_specfiles */
 
 #define ALL_SYSCALLS32 \
-    SYSCALL_ENTRY( 0x0000, NtAcceptConnectPort, 24 ) \
-    SYSCALL_ENTRY( 0x0001, NtAccessCheck, 32 ) \
-    SYSCALL_ENTRY( 0x0002, NtAccessCheckAndAuditAlarm, 44 ) \
-    SYSCALL_ENTRY( 0x0003, NtAddAtom, 12 ) \
-    SYSCALL_ENTRY( 0x0004, NtAdjustGroupsToken, 24 ) \
-    SYSCALL_ENTRY( 0x0005, NtAdjustPrivilegesToken, 24 ) \
-    SYSCALL_ENTRY( 0x0006, NtAlertMultipleThreadByThreadId, 16 ) \
-    SYSCALL_ENTRY( 0x0007, NtAlertResumeThread, 8 ) \
-    SYSCALL_ENTRY( 0x0008, NtAlertThread, 4 ) \
-    SYSCALL_ENTRY( 0x0009, NtAlertThreadByThreadId, 4 ) \
-    SYSCALL_ENTRY( 0x000a, NtAllocateLocallyUniqueId, 4 ) \
-    SYSCALL_ENTRY( 0x000b, NtAllocateReserveObject, 12 ) \
-    SYSCALL_ENTRY( 0x000c, NtAllocateUuids, 16 ) \
-    SYSCALL_ENTRY( 0x000d, NtAllocateVirtualMemory, 24 ) \
-    SYSCALL_ENTRY( 0x000e, NtAllocateVirtualMemoryEx, 28 ) \
-    SYSCALL_ENTRY( 0x000f, NtAreMappedFilesTheSame, 8 ) \
-    SYSCALL_ENTRY( 0x0010, NtAssignProcessToJobObject, 8 ) \
-    SYSCALL_ENTRY( 0x0011, NtCallbackReturn, 12 ) \
-    SYSCALL_ENTRY( 0x0012, NtCancelIoFile, 8 ) \
-    SYSCALL_ENTRY( 0x0013, NtCancelIoFileEx, 12 ) \
-    SYSCALL_ENTRY( 0x0014, NtCancelSynchronousIoFile, 12 ) \
-    SYSCALL_ENTRY( 0x0015, NtCancelTimer, 8 ) \
-    SYSCALL_ENTRY( 0x0016, NtClearEvent, 4 ) \
-    SYSCALL_ENTRY( 0x0017, NtClose, 4 ) \
-    SYSCALL_ENTRY( 0x0018, NtCommitTransaction, 8 ) \
-    SYSCALL_ENTRY( 0x0019, NtCompareObjects, 8 ) \
-    SYSCALL_ENTRY( 0x001a, NtCompareTokens, 12 ) \
-    SYSCALL_ENTRY( 0x001b, NtCompleteConnectPort, 4 ) \
-    SYSCALL_ENTRY( 0x001c, NtConnectPort, 32 ) \
-    SYSCALL_ENTRY( 0x001d, NtContinue, 8 ) \
-    SYSCALL_ENTRY( 0x001e, NtContinueEx, 8 ) \
-    SYSCALL_ENTRY( 0x001f, NtConvertBetweenAuxiliaryCounterAndPerformanceCounter, 16 ) \
-    SYSCALL_ENTRY( 0x0020, NtCreateDebugObject, 16 ) \
-    SYSCALL_ENTRY( 0x0021, NtCreateDirectoryObject, 12 ) \
-    SYSCALL_ENTRY( 0x0022, NtCreateEvent, 20 ) \
-    SYSCALL_ENTRY( 0x0023, NtCreateFile, 44 ) \
-    SYSCALL_ENTRY( 0x0024, NtCreateIoCompletion, 16 ) \
-    SYSCALL_ENTRY( 0x0025, NtCreateJobObject, 12 ) \
-    SYSCALL_ENTRY( 0x0026, NtCreateKey, 28 ) \
-    SYSCALL_ENTRY( 0x0027, NtCreateKeyTransacted, 32 ) \
-    SYSCALL_ENTRY( 0x0028, NtCreateKeyedEvent, 16 ) \
-    SYSCALL_ENTRY( 0x0029, NtCreateLowBoxToken, 36 ) \
-    SYSCALL_ENTRY( 0x002a, NtCreateMailslotFile, 32 ) \
-    SYSCALL_ENTRY( 0x002b, NtCreateMutant, 16 ) \
-    SYSCALL_ENTRY( 0x002c, NtCreateNamedPipeFile, 56 ) \
-    SYSCALL_ENTRY( 0x002d, NtCreatePagingFile, 16 ) \
-    SYSCALL_ENTRY( 0x002e, NtCreatePort, 20 ) \
-    SYSCALL_ENTRY( 0x002f, NtCreateSection, 28 ) \
-    SYSCALL_ENTRY( 0x0030, NtCreateSectionEx, 36 ) \
-    SYSCALL_ENTRY( 0x0031, NtCreateSemaphore, 20 ) \
-    SYSCALL_ENTRY( 0x0032, NtCreateSymbolicLinkObject, 16 ) \
-    SYSCALL_ENTRY( 0x0033, NtCreateThread, 32 ) \
-    SYSCALL_ENTRY( 0x0034, NtCreateThreadEx, 44 ) \
-    SYSCALL_ENTRY( 0x0035, NtCreateTimer, 16 ) \
-    SYSCALL_ENTRY( 0x0036, NtCreateToken, 52 ) \
-    SYSCALL_ENTRY( 0x0037, NtCreateTransaction, 40 ) \
-    SYSCALL_ENTRY( 0x0038, NtCreateUserProcess, 44 ) \
-    SYSCALL_ENTRY( 0x0039, NtDebugActiveProcess, 8 ) \
-    SYSCALL_ENTRY( 0x003a, NtDebugContinue, 12 ) \
-    SYSCALL_ENTRY( 0x003b, NtDelayExecution, 8 ) \
-    SYSCALL_ENTRY( 0x003c, NtDeleteAtom, 4 ) \
-    SYSCALL_ENTRY( 0x003d, NtDeleteFile, 4 ) \
-    SYSCALL_ENTRY( 0x003e, NtDeleteKey, 4 ) \
-    SYSCALL_ENTRY( 0x003f, NtDeleteValueKey, 8 ) \
-    SYSCALL_ENTRY( 0x0040, NtDeviceIoControlFile, 40 ) \
-    SYSCALL_ENTRY( 0x0041, NtDisplayString, 4 ) \
-    SYSCALL_ENTRY( 0x0042, NtDuplicateObject, 28 ) \
-    SYSCALL_ENTRY( 0x0043, NtDuplicateToken, 24 ) \
-    SYSCALL_ENTRY( 0x0044, NtEnumerateKey, 24 ) \
-    SYSCALL_ENTRY( 0x0045, NtEnumerateValueKey, 24 ) \
-    SYSCALL_ENTRY( 0x0046, NtFilterToken, 24 ) \
-    SYSCALL_ENTRY( 0x0047, NtFindAtom, 12 ) \
-    SYSCALL_ENTRY( 0x0048, NtFlushBuffersFile, 8 ) \
-    SYSCALL_ENTRY( 0x0049, NtFlushBuffersFileEx, 20 ) \
-    SYSCALL_ENTRY( 0x004a, NtFlushInstructionCache, 12 ) \
-    SYSCALL_ENTRY( 0x004b, NtFlushKey, 4 ) \
-    SYSCALL_ENTRY( 0x004c, NtFlushProcessWriteBuffers, 0 ) \
-    SYSCALL_ENTRY( 0x004d, NtFlushVirtualMemory, 16 ) \
-    SYSCALL_ENTRY( 0x004e, NtFreeVirtualMemory, 16 ) \
-    SYSCALL_ENTRY( 0x004f, NtFsControlFile, 40 ) \
-    SYSCALL_ENTRY( 0x0050, NtGetContextThread, 8 ) \
-    SYSCALL_ENTRY( 0x0051, NtGetCurrentProcessorNumber, 0 ) \
-    SYSCALL_ENTRY( 0x0052, NtGetNextProcess, 20 ) \
-    SYSCALL_ENTRY( 0x0053, NtGetNextThread, 24 ) \
-    SYSCALL_ENTRY( 0x0054, NtGetNlsSectionPtr, 20 ) \
-    SYSCALL_ENTRY( 0x0055, NtGetWriteWatch, 28 ) \
-    SYSCALL_ENTRY( 0x0056, NtImpersonateAnonymousToken, 4 ) \
-    SYSCALL_ENTRY( 0x0057, NtInitializeNlsFiles, 12 ) \
-    SYSCALL_ENTRY( 0x0058, NtInitiatePowerAction, 16 ) \
-    SYSCALL_ENTRY( 0x0059, NtIsProcessInJob, 8 ) \
-    SYSCALL_ENTRY( 0x005a, NtListenPort, 8 ) \
-    SYSCALL_ENTRY( 0x005b, NtLoadDriver, 4 ) \
-    SYSCALL_ENTRY( 0x005c, NtLoadKey, 8 ) \
-    SYSCALL_ENTRY( 0x005d, NtLoadKey2, 12 ) \
-    SYSCALL_ENTRY( 0x005e, NtLoadKeyEx, 32 ) \
-    SYSCALL_ENTRY( 0x005f, NtLockFile, 40 ) \
-    SYSCALL_ENTRY( 0x0060, NtLockVirtualMemory, 16 ) \
-    SYSCALL_ENTRY( 0x0061, NtMakePermanentObject, 4 ) \
-    SYSCALL_ENTRY( 0x0062, NtMakeTemporaryObject, 4 ) \
-    SYSCALL_ENTRY( 0x0063, NtMapViewOfSection, 40 ) \
-    SYSCALL_ENTRY( 0x0064, NtMapViewOfSectionEx, 36 ) \
-    SYSCALL_ENTRY( 0x0065, NtNotifyChangeDirectoryFile, 36 ) \
-    SYSCALL_ENTRY( 0x0066, NtNotifyChangeKey, 40 ) \
-    SYSCALL_ENTRY( 0x0067, NtNotifyChangeMultipleKeys, 48 ) \
-    SYSCALL_ENTRY( 0x0068, NtOpenDirectoryObject, 12 ) \
-    SYSCALL_ENTRY( 0x0069, NtOpenEvent, 12 ) \
-    SYSCALL_ENTRY( 0x006a, NtOpenFile, 24 ) \
-    SYSCALL_ENTRY( 0x006b, NtOpenIoCompletion, 12 ) \
-    SYSCALL_ENTRY( 0x006c, NtOpenJobObject, 12 ) \
-    SYSCALL_ENTRY( 0x006d, NtOpenKey, 12 ) \
-    SYSCALL_ENTRY( 0x006e, NtOpenKeyEx, 16 ) \
-    SYSCALL_ENTRY( 0x006f, NtOpenKeyTransacted, 16 ) \
-    SYSCALL_ENTRY( 0x0070, NtOpenKeyTransactedEx, 20 ) \
-    SYSCALL_ENTRY( 0x0071, NtOpenKeyedEvent, 12 ) \
-    SYSCALL_ENTRY( 0x0072, NtOpenMutant, 12 ) \
-    SYSCALL_ENTRY( 0x0073, NtOpenProcess, 16 ) \
-    SYSCALL_ENTRY( 0x0074, NtOpenProcessToken, 12 ) \
-    SYSCALL_ENTRY( 0x0075, NtOpenProcessTokenEx, 16 ) \
-    SYSCALL_ENTRY( 0x0076, NtOpenSection, 12 ) \
-    SYSCALL_ENTRY( 0x0077, NtOpenSemaphore, 12 ) \
-    SYSCALL_ENTRY( 0x0078, NtOpenSymbolicLinkObject, 12 ) \
-    SYSCALL_ENTRY( 0x0079, NtOpenThread, 16 ) \
-    SYSCALL_ENTRY( 0x007a, NtOpenThreadToken, 16 ) \
-    SYSCALL_ENTRY( 0x007b, NtOpenThreadTokenEx, 20 ) \
-    SYSCALL_ENTRY( 0x007c, NtOpenTimer, 12 ) \
-    SYSCALL_ENTRY( 0x007d, NtPowerInformation, 20 ) \
-    SYSCALL_ENTRY( 0x007e, NtPrivilegeCheck, 12 ) \
-    SYSCALL_ENTRY( 0x007f, NtProtectVirtualMemory, 20 ) \
-    SYSCALL_ENTRY( 0x0080, NtPulseEvent, 8 ) \
-    SYSCALL_ENTRY( 0x0081, NtQueryAttributesFile, 8 ) \
-    SYSCALL_ENTRY( 0x0082, NtQueryDefaultLocale, 8 ) \
-    SYSCALL_ENTRY( 0x0083, NtQueryDefaultUILanguage, 4 ) \
-    SYSCALL_ENTRY( 0x0084, NtQueryDirectoryFile, 44 ) \
-    SYSCALL_ENTRY( 0x0085, NtQueryDirectoryObject, 28 ) \
-    SYSCALL_ENTRY( 0x0086, NtQueryEaFile, 36 ) \
-    SYSCALL_ENTRY( 0x0087, NtQueryEvent, 20 ) \
-    SYSCALL_ENTRY( 0x0088, NtQueryFullAttributesFile, 8 ) \
-    SYSCALL_ENTRY( 0x0089, NtQueryInformationAtom, 20 ) \
-    SYSCALL_ENTRY( 0x008a, NtQueryInformationFile, 20 ) \
-    SYSCALL_ENTRY( 0x008b, NtQueryInformationJobObject, 20 ) \
-    SYSCALL_ENTRY_NtQueryInformationProcess( 0x008c, NtQueryInformationProcess, 20 ) \
-    SYSCALL_ENTRY( 0x008d, NtQueryInformationThread, 20 ) \
-    SYSCALL_ENTRY( 0x008e, NtQueryInformationToken, 20 ) \
-    SYSCALL_ENTRY( 0x008f, NtQueryInstallUILanguage, 4 ) \
-    SYSCALL_ENTRY( 0x0090, NtQueryIoCompletion, 20 ) \
-    SYSCALL_ENTRY( 0x0091, NtQueryKey, 20 ) \
-    SYSCALL_ENTRY( 0x0092, NtQueryLicenseValue, 20 ) \
-    SYSCALL_ENTRY( 0x0093, NtQueryMultipleValueKey, 24 ) \
-    SYSCALL_ENTRY( 0x0094, NtQueryMutant, 20 ) \
-    SYSCALL_ENTRY( 0x0095, NtQueryObject, 20 ) \
-    SYSCALL_ENTRY( 0x0096, NtQueryPerformanceCounter, 8 ) \
-    SYSCALL_ENTRY( 0x0097, NtQuerySection, 20 ) \
-    SYSCALL_ENTRY( 0x0098, NtQuerySecurityObject, 20 ) \
-    SYSCALL_ENTRY( 0x0099, NtQuerySemaphore, 20 ) \
-    SYSCALL_ENTRY( 0x009a, NtQuerySymbolicLinkObject, 12 ) \
-    SYSCALL_ENTRY( 0x009b, NtQuerySystemEnvironmentValue, 16 ) \
-    SYSCALL_ENTRY( 0x009c, NtQuerySystemEnvironmentValueEx, 20 ) \
-    SYSCALL_ENTRY( 0x009d, NtQuerySystemInformation, 16 ) \
-    SYSCALL_ENTRY( 0x009e, NtQuerySystemInformationEx, 24 ) \
-    SYSCALL_ENTRY_NtQuerySystemTime( 0x009f, NtQuerySystemTime, 4 ) \
-    SYSCALL_ENTRY( 0x00a0, NtQueryTimer, 20 ) \
-    SYSCALL_ENTRY( 0x00a1, NtQueryTimerResolution, 12 ) \
-    SYSCALL_ENTRY( 0x00a2, NtQueryValueKey, 24 ) \
-    SYSCALL_ENTRY( 0x00a3, NtQueryVirtualMemory, 24 ) \
-    SYSCALL_ENTRY( 0x00a4, NtQueryVolumeInformationFile, 20 ) \
-    SYSCALL_ENTRY( 0x00a5, NtQueueApcThread, 20 ) \
-    SYSCALL_ENTRY( 0x00a6, NtQueueApcThreadEx, 24 ) \
-    SYSCALL_ENTRY( 0x00a7, NtRaiseException, 12 ) \
-    SYSCALL_ENTRY( 0x00a8, NtRaiseHardError, 24 ) \
-    SYSCALL_ENTRY( 0x00a9, NtReadFile, 36 ) \
-    SYSCALL_ENTRY( 0x00aa, NtReadFileScatter, 36 ) \
-    SYSCALL_ENTRY( 0x00ab, NtReadVirtualMemory, 20 ) \
-    SYSCALL_ENTRY( 0x00ac, NtRegisterThreadTerminatePort, 4 ) \
-    SYSCALL_ENTRY( 0x00ad, NtReleaseKeyedEvent, 16 ) \
-    SYSCALL_ENTRY( 0x00ae, NtReleaseMutant, 8 ) \
-    SYSCALL_ENTRY( 0x00af, NtReleaseSemaphore, 12 ) \
-    SYSCALL_ENTRY( 0x00b0, NtRemoveIoCompletion, 20 ) \
-    SYSCALL_ENTRY( 0x00b1, NtRemoveIoCompletionEx, 24 ) \
-    SYSCALL_ENTRY( 0x00b2, NtRemoveProcessDebug, 8 ) \
-    SYSCALL_ENTRY( 0x00b3, NtRenameKey, 8 ) \
-    SYSCALL_ENTRY( 0x00b4, NtReplaceKey, 12 ) \
-    SYSCALL_ENTRY( 0x00b5, NtReplyWaitReceivePort, 16 ) \
-    SYSCALL_ENTRY( 0x00b6, NtRequestWaitReplyPort, 12 ) \
-    SYSCALL_ENTRY( 0x00b7, NtResetEvent, 8 ) \
-    SYSCALL_ENTRY( 0x00b8, NtResetWriteWatch, 12 ) \
-    SYSCALL_ENTRY( 0x00b9, NtRestoreKey, 12 ) \
-    SYSCALL_ENTRY( 0x00ba, NtResumeProcess, 4 ) \
-    SYSCALL_ENTRY( 0x00bb, NtResumeThread, 8 ) \
-    SYSCALL_ENTRY( 0x00bc, NtRollbackTransaction, 8 ) \
-    SYSCALL_ENTRY( 0x00bd, NtSaveKey, 8 ) \
-    SYSCALL_ENTRY( 0x00be, NtSecureConnectPort, 36 ) \
-    SYSCALL_ENTRY( 0x00bf, NtSetContextThread, 8 ) \
-    SYSCALL_ENTRY( 0x00c0, NtSetDebugFilterState, 12 ) \
-    SYSCALL_ENTRY( 0x00c1, NtSetDefaultLocale, 8 ) \
-    SYSCALL_ENTRY( 0x00c2, NtSetDefaultUILanguage, 4 ) \
-    SYSCALL_ENTRY( 0x00c3, NtSetEaFile, 16 ) \
-    SYSCALL_ENTRY( 0x00c4, NtSetEvent, 8 ) \
-    SYSCALL_ENTRY( 0x00c5, NtSetInformationDebugObject, 20 ) \
-    SYSCALL_ENTRY( 0x00c6, NtSetInformationFile, 20 ) \
-    SYSCALL_ENTRY( 0x00c7, NtSetInformationJobObject, 16 ) \
-    SYSCALL_ENTRY( 0x00c8, NtSetInformationKey, 16 ) \
-    SYSCALL_ENTRY( 0x00c9, NtSetInformationObject, 16 ) \
-    SYSCALL_ENTRY( 0x00ca, NtSetInformationProcess, 16 ) \
-    SYSCALL_ENTRY( 0x00cb, NtSetInformationThread, 16 ) \
-    SYSCALL_ENTRY( 0x00cc, NtSetInformationToken, 16 ) \
-    SYSCALL_ENTRY( 0x00cd, NtSetInformationVirtualMemory, 24 ) \
-    SYSCALL_ENTRY( 0x00ce, NtSetIntervalProfile, 8 ) \
-    SYSCALL_ENTRY( 0x00cf, NtSetIoCompletion, 20 ) \
-    SYSCALL_ENTRY( 0x00d0, NtSetIoCompletionEx, 24 ) \
-    SYSCALL_ENTRY( 0x00d1, NtSetLdtEntries, 24 ) \
-    SYSCALL_ENTRY( 0x00d2, NtSetSecurityObject, 12 ) \
-    SYSCALL_ENTRY( 0x00d3, NtSetSystemInformation, 12 ) \
-    SYSCALL_ENTRY( 0x00d4, NtSetSystemTime, 8 ) \
-    SYSCALL_ENTRY( 0x00d5, NtSetThreadExecutionState, 8 ) \
-    SYSCALL_ENTRY( 0x00d6, NtSetTimer, 28 ) \
-    SYSCALL_ENTRY( 0x00d7, NtSetTimerResolution, 12 ) \
-    SYSCALL_ENTRY( 0x00d8, NtSetValueKey, 24 ) \
-    SYSCALL_ENTRY( 0x00d9, NtSetVolumeInformationFile, 20 ) \
-    SYSCALL_ENTRY( 0x00da, NtShutdownSystem, 4 ) \
-    SYSCALL_ENTRY( 0x00db, NtSignalAndWaitForSingleObject, 16 ) \
-    SYSCALL_ENTRY( 0x00dc, NtSuspendProcess, 4 ) \
-    SYSCALL_ENTRY( 0x00dd, NtSuspendThread, 8 ) \
-    SYSCALL_ENTRY( 0x00de, NtSystemDebugControl, 24 ) \
-    SYSCALL_ENTRY( 0x00df, NtTerminateJobObject, 8 ) \
-    SYSCALL_ENTRY( 0x00e0, NtTerminateProcess, 8 ) \
-    SYSCALL_ENTRY( 0x00e1, NtTerminateThread, 8 ) \
-    SYSCALL_ENTRY( 0x00e2, NtTestAlert, 0 ) \
-    SYSCALL_ENTRY( 0x00e3, NtTraceControl, 24 ) \
-    SYSCALL_ENTRY( 0x00e4, NtUnloadDriver, 4 ) \
-    SYSCALL_ENTRY( 0x00e5, NtUnloadKey, 4 ) \
-    SYSCALL_ENTRY( 0x00e6, NtUnlockFile, 20 ) \
-    SYSCALL_ENTRY( 0x00e7, NtUnlockVirtualMemory, 16 ) \
-    SYSCALL_ENTRY( 0x00e8, NtUnmapViewOfSection, 8 ) \
-    SYSCALL_ENTRY( 0x00e9, NtUnmapViewOfSectionEx, 12 ) \
-    SYSCALL_ENTRY( 0x00ea, NtWaitForAlertByThreadId, 8 ) \
-    SYSCALL_ENTRY( 0x00eb, NtWaitForDebugEvent, 16 ) \
-    SYSCALL_ENTRY( 0x00ec, NtWaitForKeyedEvent, 16 ) \
-    SYSCALL_ENTRY( 0x00ed, NtWaitForMultipleObjects, 20 ) \
-    SYSCALL_ENTRY( 0x00ee, NtWaitForSingleObject, 12 ) \
-    SYSCALL_ENTRY( 0x00ef, NtWow64AllocateVirtualMemory64, 28 ) \
-    SYSCALL_ENTRY( 0x00f0, NtWow64GetNativeSystemInformation, 16 ) \
-    SYSCALL_ENTRY( 0x00f1, NtWow64IsProcessorFeaturePresent, 4 ) \
-    SYSCALL_ENTRY( 0x00f2, NtWow64QueryInformationProcess64, 20 ) \
-    SYSCALL_ENTRY( 0x00f3, NtWow64ReadVirtualMemory64, 28 ) \
-    SYSCALL_ENTRY( 0x00f4, NtWow64WriteVirtualMemory64, 28 ) \
-    SYSCALL_ENTRY( 0x00f5, NtWriteFile, 36 ) \
-    SYSCALL_ENTRY( 0x00f6, NtWriteFileGather, 36 ) \
-    SYSCALL_ENTRY( 0x00f7, NtWriteVirtualMemory, 20 ) \
-    SYSCALL_ENTRY( 0x00f8, NtYieldExecution, 0 ) \
+    SYSCALL_ENTRY( 0x0000, NtAccessCheck, 32 ) \
+    SYSCALL_ENTRY( 0x0001, NtAdjustGroupsToken, 24 ) \
+    SYSCALL_ENTRY( 0x0002, NtAcceptConnectPort, 24 ) \
+    SYSCALL_ENTRY( 0x0003, NtAlertMultipleThreadByThreadId, 16 ) \
+    SYSCALL_ENTRY( 0x0004, NtWaitForSingleObject, 12 ) \
+    SYSCALL_ENTRY( 0x0005, NtCallbackReturn, 12 ) \
+    SYSCALL_ENTRY( 0x0006, NtReadFile, 36 ) \
+    SYSCALL_ENTRY( 0x0007, NtDeviceIoControlFile, 40 ) \
+    SYSCALL_ENTRY( 0x0008, NtWriteFile, 36 ) \
+    SYSCALL_ENTRY( 0x0009, NtRemoveIoCompletion, 20 ) \
+    SYSCALL_ENTRY( 0x000a, NtReleaseSemaphore, 12 ) \
+    SYSCALL_ENTRY( 0x000b, NtReplyWaitReceivePort, 16 ) \
+    SYSCALL_ENTRY( 0x000c, NtAlertResumeThread, 8 ) \
+    SYSCALL_ENTRY( 0x000d, NtSetInformationThread, 16 ) \
+    SYSCALL_ENTRY( 0x000e, NtSetEvent, 8 ) \
+    SYSCALL_ENTRY( 0x000f, NtClose, 4 ) \
+    SYSCALL_ENTRY( 0x0010, NtQueryObject, 20 ) \
+    SYSCALL_ENTRY( 0x0011, NtQueryInformationFile, 20 ) \
+    SYSCALL_ENTRY( 0x0012, NtOpenKey, 12 ) \
+    SYSCALL_ENTRY( 0x0013, NtEnumerateValueKey, 24 ) \
+    SYSCALL_ENTRY( 0x0014, NtFindAtom, 12 ) \
+    SYSCALL_ENTRY( 0x0015, NtQueryDefaultLocale, 8 ) \
+    SYSCALL_ENTRY( 0x0016, NtQueryKey, 20 ) \
+    SYSCALL_ENTRY( 0x0017, NtQueryValueKey, 24 ) \
+    SYSCALL_ENTRY( 0x0018, NtAllocateVirtualMemory, 24 ) \
+    SYSCALL_ENTRY_NtQueryInformationProcess( 0x0019, NtQueryInformationProcess, 20 ) \
+    SYSCALL_ENTRY( 0x001a, NtAlertThread, 4 ) \
+    SYSCALL_ENTRY( 0x001b, NtWriteFileGather, 36 ) \
+    SYSCALL_ENTRY( 0x001c, NtSetInformationProcess, 16 ) \
+    SYSCALL_ENTRY( 0x001d, NtCreateKey, 28 ) \
+    SYSCALL_ENTRY( 0x001e, NtFreeVirtualMemory, 16 ) \
+    SYSCALL_ENTRY( 0x001f, NtAlertThreadByThreadId, 4 ) \
+    SYSCALL_ENTRY( 0x0020, NtReleaseMutant, 8 ) \
+    SYSCALL_ENTRY( 0x0021, NtQueryInformationToken, 20 ) \
+    SYSCALL_ENTRY( 0x0022, NtRequestWaitReplyPort, 12 ) \
+    SYSCALL_ENTRY( 0x0023, NtQueryVirtualMemory, 24 ) \
+    SYSCALL_ENTRY( 0x0024, NtOpenThreadToken, 16 ) \
+    SYSCALL_ENTRY( 0x0025, NtQueryInformationThread, 20 ) \
+    SYSCALL_ENTRY( 0x0026, NtOpenProcess, 16 ) \
+    SYSCALL_ENTRY( 0x0027, NtSetInformationFile, 20 ) \
+    SYSCALL_ENTRY( 0x0028, NtMapViewOfSection, 40 ) \
+    SYSCALL_ENTRY( 0x0029, NtAccessCheckAndAuditAlarm, 44 ) \
+    SYSCALL_ENTRY( 0x002a, NtUnmapViewOfSection, 8 ) \
+    SYSCALL_ENTRY( 0x002b, NtAllocateLocallyUniqueId, 4 ) \
+    SYSCALL_ENTRY( 0x002c, NtTerminateProcess, 8 ) \
+    SYSCALL_ENTRY( 0x002d, NtAllocateReserveObject, 12 ) \
+    SYSCALL_ENTRY( 0x002e, NtReadFileScatter, 36 ) \
+    SYSCALL_ENTRY( 0x002f, NtOpenThreadTokenEx, 20 ) \
+    SYSCALL_ENTRY( 0x0030, NtOpenProcessTokenEx, 16 ) \
+    SYSCALL_ENTRY( 0x0031, NtQueryPerformanceCounter, 8 ) \
+    SYSCALL_ENTRY( 0x0032, NtEnumerateKey, 24 ) \
+    SYSCALL_ENTRY( 0x0033, NtOpenFile, 24 ) \
+    SYSCALL_ENTRY( 0x0034, NtDelayExecution, 8 ) \
+    SYSCALL_ENTRY( 0x0035, NtQueryDirectoryFile, 44 ) \
+    SYSCALL_ENTRY( 0x0036, NtQuerySystemInformation, 16 ) \
+    SYSCALL_ENTRY( 0x0037, NtOpenSection, 12 ) \
+    SYSCALL_ENTRY( 0x0038, NtQueryTimer, 20 ) \
+    SYSCALL_ENTRY( 0x0039, NtFsControlFile, 40 ) \
+    SYSCALL_ENTRY( 0x003a, NtWriteVirtualMemory, 20 ) \
+    SYSCALL_ENTRY( 0x003b, NtAllocateUuids, 16 ) \
+    SYSCALL_ENTRY( 0x003c, NtDuplicateObject, 28 ) \
+    SYSCALL_ENTRY( 0x003d, NtQueryAttributesFile, 8 ) \
+    SYSCALL_ENTRY( 0x003e, NtClearEvent, 4 ) \
+    SYSCALL_ENTRY( 0x003f, NtReadVirtualMemory, 20 ) \
+    SYSCALL_ENTRY( 0x0040, NtOpenEvent, 12 ) \
+    SYSCALL_ENTRY( 0x0041, NtAdjustPrivilegesToken, 24 ) \
+    SYSCALL_ENTRY( 0x0042, NtDuplicateToken, 24 ) \
+    SYSCALL_ENTRY( 0x0043, NtContinue, 8 ) \
+    SYSCALL_ENTRY( 0x0044, NtQueryDefaultUILanguage, 4 ) \
+    SYSCALL_ENTRY( 0x0045, NtQueueApcThread, 20 ) \
+    SYSCALL_ENTRY( 0x0046, NtYieldExecution, 0 ) \
+    SYSCALL_ENTRY( 0x0047, NtAddAtom, 12 ) \
+    SYSCALL_ENTRY( 0x0048, NtCreateEvent, 20 ) \
+    SYSCALL_ENTRY( 0x0049, NtQueryVolumeInformationFile, 20 ) \
+    SYSCALL_ENTRY( 0x004a, NtCreateSection, 28 ) \
+    SYSCALL_ENTRY( 0x004b, NtFlushBuffersFile, 8 ) \
+    SYSCALL_ENTRY( 0x004c, NtAllocateVirtualMemoryEx, 28 ) \
+    SYSCALL_ENTRY( 0x004d, NtAreMappedFilesTheSame, 8 ) \
+    SYSCALL_ENTRY( 0x004e, NtCreateThread, 32 ) \
+    SYSCALL_ENTRY( 0x004f, NtIsProcessInJob, 8 ) \
+    SYSCALL_ENTRY( 0x0050, NtProtectVirtualMemory, 20 ) \
+    SYSCALL_ENTRY( 0x0051, NtQuerySection, 20 ) \
+    SYSCALL_ENTRY( 0x0052, NtResumeThread, 8 ) \
+    SYSCALL_ENTRY( 0x0053, NtTerminateThread, 8 ) \
+    SYSCALL_ENTRY( 0x0054, NtAssignProcessToJobObject, 8 ) \
+    SYSCALL_ENTRY( 0x0055, NtCreateFile, 44 ) \
+    SYSCALL_ENTRY( 0x0056, NtQueryEvent, 20 ) \
+    SYSCALL_ENTRY( 0x0057, NtCancelIoFileEx, 12 ) \
+    SYSCALL_ENTRY( 0x0058, NtOpenDirectoryObject, 12 ) \
+    SYSCALL_ENTRY( 0x0059, NtCancelSynchronousIoFile, 12 ) \
+    SYSCALL_ENTRY_NtQuerySystemTime( 0x005a, NtQuerySystemTime, 4 ) \
+    SYSCALL_ENTRY( 0x005b, NtWaitForMultipleObjects, 20 ) \
+    SYSCALL_ENTRY( 0x005c, NtSetInformationObject, 16 ) \
+    SYSCALL_ENTRY( 0x005d, NtCancelIoFile, 8 ) \
+    SYSCALL_ENTRY( 0x005e, NtCommitTransaction, 8 ) \
+    SYSCALL_ENTRY( 0x005f, NtPowerInformation, 20 ) \
+    SYSCALL_ENTRY( 0x0060, NtSetValueKey, 24 ) \
+    SYSCALL_ENTRY( 0x0061, NtCancelTimer, 8 ) \
+    SYSCALL_ENTRY( 0x0062, NtSetTimer, 28 ) \
+    SYSCALL_ENTRY( 0x0063, NtCompareObjects, 8 ) \
+    SYSCALL_ENTRY( 0x0064, NtCompareTokens, 12 ) \
+    SYSCALL_ENTRY( 0x0065, NtCompleteConnectPort, 4 ) \
+    SYSCALL_ENTRY( 0x0066, NtConnectPort, 32 ) \
+    SYSCALL_ENTRY( 0x0067, NtContinueEx, 8 ) \
+    SYSCALL_ENTRY( 0x0068, NtConvertBetweenAuxiliaryCounterAndPerformanceCounter, 16 ) \
+    SYSCALL_ENTRY( 0x0069, NtCreateDirectoryObject, 12 ) \
+    SYSCALL_ENTRY( 0x006a, NtCreateIoCompletion, 16 ) \
+    SYSCALL_ENTRY( 0x006b, NtCreateJobObject, 12 ) \
+    SYSCALL_ENTRY( 0x006c, NtCreateKeyTransacted, 32 ) \
+    SYSCALL_ENTRY( 0x006d, NtCreateKeyedEvent, 16 ) \
+    SYSCALL_ENTRY( 0x006e, NtCreateLowBoxToken, 36 ) \
+    SYSCALL_ENTRY( 0x006f, NtCreateMailslotFile, 32 ) \
+    SYSCALL_ENTRY( 0x0070, NtCreateMutant, 16 ) \
+    SYSCALL_ENTRY( 0x0071, NtCreateNamedPipeFile, 56 ) \
+    SYSCALL_ENTRY( 0x0072, NtCreatePagingFile, 16 ) \
+    SYSCALL_ENTRY( 0x0073, NtCreatePort, 20 ) \
+    SYSCALL_ENTRY( 0x0074, NtCreateSectionEx, 36 ) \
+    SYSCALL_ENTRY( 0x0075, NtCreateSemaphore, 20 ) \
+    SYSCALL_ENTRY( 0x0076, NtCreateSymbolicLinkObject, 16 ) \
+    SYSCALL_ENTRY( 0x0077, NtCreateThreadEx, 44 ) \
+    SYSCALL_ENTRY( 0x0078, NtCreateTimer, 16 ) \
+    SYSCALL_ENTRY( 0x0079, NtCreateToken, 52 ) \
+    SYSCALL_ENTRY( 0x007a, NtCreateTransaction, 40 ) \
+    SYSCALL_ENTRY( 0x007b, NtCreateUserProcess, 44 ) \
+    SYSCALL_ENTRY( 0x007c, NtDebugActiveProcess, 8 ) \
+    SYSCALL_ENTRY( 0x007d, NtDebugContinue, 12 ) \
+    SYSCALL_ENTRY( 0x007e, NtDeleteAtom, 4 ) \
+    SYSCALL_ENTRY( 0x007f, NtDeleteFile, 4 ) \
+    SYSCALL_ENTRY( 0x0080, NtDeleteKey, 4 ) \
+    SYSCALL_ENTRY( 0x0081, NtDeleteValueKey, 8 ) \
+    SYSCALL_ENTRY( 0x0082, NtDisplayString, 4 ) \
+    SYSCALL_ENTRY( 0x0083, NtFilterToken, 24 ) \
+    SYSCALL_ENTRY( 0x0084, NtFlushBuffersFileEx, 20 ) \
+    SYSCALL_ENTRY( 0x0085, NtFlushInstructionCache, 12 ) \
+    SYSCALL_ENTRY( 0x0086, NtFlushKey, 4 ) \
+    SYSCALL_ENTRY( 0x0087, NtFlushProcessWriteBuffers, 0 ) \
+    SYSCALL_ENTRY( 0x0088, NtFlushVirtualMemory, 16 ) \
+    SYSCALL_ENTRY( 0x0089, NtGetContextThread, 8 ) \
+    SYSCALL_ENTRY( 0x008a, NtGetCurrentProcessorNumber, 0 ) \
+    SYSCALL_ENTRY( 0x008b, NtGetNextProcess, 20 ) \
+    SYSCALL_ENTRY( 0x008c, NtGetNextThread, 24 ) \
+    SYSCALL_ENTRY( 0x008d, NtGetNlsSectionPtr, 20 ) \
+    SYSCALL_ENTRY( 0x008e, NtGetWriteWatch, 28 ) \
+    SYSCALL_ENTRY( 0x008f, NtImpersonateAnonymousToken, 4 ) \
+    SYSCALL_ENTRY( 0x0090, NtInitializeNlsFiles, 12 ) \
+    SYSCALL_ENTRY( 0x0091, NtInitiatePowerAction, 16 ) \
+    SYSCALL_ENTRY( 0x0092, NtListenPort, 8 ) \
+    SYSCALL_ENTRY( 0x0093, NtLoadDriver, 4 ) \
+    SYSCALL_ENTRY( 0x0094, NtLoadKey, 8 ) \
+    SYSCALL_ENTRY( 0x0095, NtLoadKey2, 12 ) \
+    SYSCALL_ENTRY( 0x0096, NtLoadKeyEx, 32 ) \
+    SYSCALL_ENTRY( 0x0097, NtLockFile, 40 ) \
+    SYSCALL_ENTRY( 0x0098, NtLockVirtualMemory, 16 ) \
+    SYSCALL_ENTRY( 0x0099, NtMakePermanentObject, 4 ) \
+    SYSCALL_ENTRY( 0x009a, NtMakeTemporaryObject, 4 ) \
+    SYSCALL_ENTRY( 0x009b, NtMapViewOfSectionEx, 36 ) \
+    SYSCALL_ENTRY( 0x009c, NtNotifyChangeDirectoryFile, 36 ) \
+    SYSCALL_ENTRY( 0x009d, NtNotifyChangeKey, 40 ) \
+    SYSCALL_ENTRY( 0x009e, NtNotifyChangeMultipleKeys, 48 ) \
+    SYSCALL_ENTRY( 0x009f, NtOpenIoCompletion, 12 ) \
+    SYSCALL_ENTRY( 0x00a0, NtOpenJobObject, 12 ) \
+    SYSCALL_ENTRY( 0x00a1, NtOpenKeyEx, 16 ) \
+    SYSCALL_ENTRY( 0x00a2, NtOpenKeyTransacted, 16 ) \
+    SYSCALL_ENTRY( 0x00a3, NtOpenKeyTransactedEx, 20 ) \
+    SYSCALL_ENTRY( 0x00a4, NtOpenKeyedEvent, 12 ) \
+    SYSCALL_ENTRY( 0x00a5, NtOpenMutant, 12 ) \
+    SYSCALL_ENTRY( 0x00a6, NtCreateDebugObject, 16 ) \
+    SYSCALL_ENTRY( 0x00a7, NtOpenProcessToken, 12 ) \
+    SYSCALL_ENTRY( 0x00a8, NtOpenSemaphore, 12 ) \
+    SYSCALL_ENTRY( 0x00a9, NtOpenSymbolicLinkObject, 12 ) \
+    SYSCALL_ENTRY( 0x00aa, NtOpenThread, 16 ) \
+    SYSCALL_ENTRY( 0x00ab, NtOpenTimer, 12 ) \
+    SYSCALL_ENTRY( 0x00ac, NtPrivilegeCheck, 12 ) \
+    SYSCALL_ENTRY( 0x00ad, NtPulseEvent, 8 ) \
+    SYSCALL_ENTRY( 0x00ae, NtQueryDirectoryObject, 28 ) \
+    SYSCALL_ENTRY( 0x00af, NtQueryEaFile, 36 ) \
+    SYSCALL_ENTRY( 0x00b0, NtQueryFullAttributesFile, 8 ) \
+    SYSCALL_ENTRY( 0x00b1, NtQueryInformationAtom, 20 ) \
+    SYSCALL_ENTRY( 0x00b2, NtQueryInformationJobObject, 20 ) \
+    SYSCALL_ENTRY( 0x00b3, NtQueryInstallUILanguage, 4 ) \
+    SYSCALL_ENTRY( 0x00b4, NtQueryIoCompletion, 20 ) \
+    SYSCALL_ENTRY( 0x00b5, NtQueryLicenseValue, 20 ) \
+    SYSCALL_ENTRY( 0x00b6, NtQueryMultipleValueKey, 24 ) \
+    SYSCALL_ENTRY( 0x00b7, NtQueryMutant, 20 ) \
+    SYSCALL_ENTRY( 0x00b8, NtQuerySecurityObject, 20 ) \
+    SYSCALL_ENTRY( 0x00b9, NtQuerySemaphore, 20 ) \
+    SYSCALL_ENTRY( 0x00ba, NtQuerySymbolicLinkObject, 12 ) \
+    SYSCALL_ENTRY( 0x00bb, NtQuerySystemEnvironmentValue, 16 ) \
+    SYSCALL_ENTRY( 0x00bc, NtQuerySystemEnvironmentValueEx, 20 ) \
+    SYSCALL_ENTRY( 0x00bd, NtQuerySystemInformationEx, 24 ) \
+    SYSCALL_ENTRY( 0x00be, NtQueryTimerResolution, 12 ) \
+    SYSCALL_ENTRY( 0x00bf, NtQueueApcThreadEx, 24 ) \
+    SYSCALL_ENTRY( 0x00c0, NtRaiseException, 12 ) \
+    SYSCALL_ENTRY( 0x00c1, NtRaiseHardError, 24 ) \
+    SYSCALL_ENTRY( 0x00c2, NtRegisterThreadTerminatePort, 4 ) \
+    SYSCALL_ENTRY( 0x00c3, NtReleaseKeyedEvent, 16 ) \
+    SYSCALL_ENTRY( 0x00c4, NtRemoveIoCompletionEx, 24 ) \
+    SYSCALL_ENTRY( 0x00c5, NtRemoveProcessDebug, 8 ) \
+    SYSCALL_ENTRY( 0x00c6, NtRenameKey, 8 ) \
+    SYSCALL_ENTRY( 0x00c7, NtReplaceKey, 12 ) \
+    SYSCALL_ENTRY( 0x00c8, NtResetEvent, 8 ) \
+    SYSCALL_ENTRY( 0x00c9, NtResetWriteWatch, 12 ) \
+    SYSCALL_ENTRY( 0x00ca, NtRestoreKey, 12 ) \
+    SYSCALL_ENTRY( 0x00cb, NtResumeProcess, 4 ) \
+    SYSCALL_ENTRY( 0x00cc, NtRollbackTransaction, 8 ) \
+    SYSCALL_ENTRY( 0x00cd, NtSaveKey, 8 ) \
+    SYSCALL_ENTRY( 0x00ce, NtSecureConnectPort, 36 ) \
+    SYSCALL_ENTRY( 0x00cf, NtSetContextThread, 8 ) \
+    SYSCALL_ENTRY( 0x00d0, NtSetDebugFilterState, 12 ) \
+    SYSCALL_ENTRY( 0x00d1, NtSetDefaultLocale, 8 ) \
+    SYSCALL_ENTRY( 0x00d2, NtSetDefaultUILanguage, 4 ) \
+    SYSCALL_ENTRY( 0x00d3, NtSetEaFile, 16 ) \
+    SYSCALL_ENTRY( 0x00d4, NtSetInformationDebugObject, 20 ) \
+    SYSCALL_ENTRY( 0x00d5, NtSetInformationJobObject, 16 ) \
+    SYSCALL_ENTRY( 0x00d6, NtSetInformationKey, 16 ) \
+    SYSCALL_ENTRY( 0x00d7, NtSetInformationToken, 16 ) \
+    SYSCALL_ENTRY( 0x00d8, NtSetInformationVirtualMemory, 24 ) \
+    SYSCALL_ENTRY( 0x00d9, NtSetIntervalProfile, 8 ) \
+    SYSCALL_ENTRY( 0x00da, NtSetIoCompletion, 20 ) \
+    SYSCALL_ENTRY( 0x00db, NtSetIoCompletionEx, 24 ) \
+    SYSCALL_ENTRY( 0x00dc, NtSetLdtEntries, 24 ) \
+    SYSCALL_ENTRY( 0x00dd, NtSetSecurityObject, 12 ) \
+    SYSCALL_ENTRY( 0x00de, NtSetSystemInformation, 12 ) \
+    SYSCALL_ENTRY( 0x00df, NtSetSystemTime, 8 ) \
+    SYSCALL_ENTRY( 0x00e0, NtSetThreadExecutionState, 8 ) \
+    SYSCALL_ENTRY( 0x00e1, NtSetTimerResolution, 12 ) \
+    SYSCALL_ENTRY( 0x00e2, NtSetVolumeInformationFile, 20 ) \
+    SYSCALL_ENTRY( 0x00e3, NtShutdownSystem, 4 ) \
+    SYSCALL_ENTRY( 0x00e4, NtSignalAndWaitForSingleObject, 16 ) \
+    SYSCALL_ENTRY( 0x00e5, NtSuspendProcess, 4 ) \
+    SYSCALL_ENTRY( 0x00e6, NtSuspendThread, 8 ) \
+    SYSCALL_ENTRY( 0x00e7, NtSystemDebugControl, 24 ) \
+    SYSCALL_ENTRY( 0x00e8, NtTerminateJobObject, 8 ) \
+    SYSCALL_ENTRY( 0x00e9, NtTestAlert, 0 ) \
+    SYSCALL_ENTRY( 0x00ea, NtTraceControl, 24 ) \
+    SYSCALL_ENTRY( 0x00eb, NtUnloadDriver, 4 ) \
+    SYSCALL_ENTRY( 0x00ec, NtUnloadKey, 4 ) \
+    SYSCALL_ENTRY( 0x00ed, NtUnlockFile, 20 ) \
+    SYSCALL_ENTRY( 0x00ee, NtUnlockVirtualMemory, 16 ) \
+    SYSCALL_ENTRY( 0x00ef, NtUnmapViewOfSectionEx, 12 ) \
+    SYSCALL_ENTRY( 0x00f0, NtWaitForAlertByThreadId, 8 ) \
+    SYSCALL_ENTRY( 0x00f1, NtWaitForDebugEvent, 16 ) \
+    SYSCALL_ENTRY( 0x00f2, NtWaitForKeyedEvent, 16 ) \
+    SYSCALL_ENTRY( 0x00f3, NtWow64AllocateVirtualMemory64, 28 ) \
+    SYSCALL_ENTRY( 0x00f4, NtWow64GetNativeSystemInformation, 16 ) \
+    SYSCALL_ENTRY( 0x00f5, NtWow64IsProcessorFeaturePresent, 4 ) \
+    SYSCALL_ENTRY( 0x00f6, NtWow64QueryInformationProcess64, 20 ) \
+    SYSCALL_ENTRY( 0x00f7, NtWow64ReadVirtualMemory64, 28 ) \
+    SYSCALL_ENTRY( 0x00f8, NtWow64WriteVirtualMemory64, 28 ) \
     SYSCALL_ENTRY( 0x00f9, wine_nt_to_unix_file_name, 16 ) \
     SYSCALL_ENTRY( 0x00fa, wine_unix_to_nt_file_name, 12 )
 #ifdef _WIN64
 #define ALL_SYSCALLS \
-    SYSCALL_ENTRY( 0x0000, NtAcceptConnectPort, 48 ) \
-    SYSCALL_ENTRY( 0x0001, NtAccessCheck, 64 ) \
-    SYSCALL_ENTRY( 0x0002, NtAccessCheckAndAuditAlarm, 88 ) \
-    SYSCALL_ENTRY( 0x0003, NtAddAtom, 24 ) \
-    SYSCALL_ENTRY( 0x0004, NtAdjustGroupsToken, 48 ) \
-    SYSCALL_ENTRY( 0x0005, NtAdjustPrivilegesToken, 48 ) \
-    SYSCALL_ENTRY( 0x0006, NtAlertMultipleThreadByThreadId, 32 ) \
-    SYSCALL_ENTRY( 0x0007, NtAlertResumeThread, 16 ) \
-    SYSCALL_ENTRY( 0x0008, NtAlertThread, 8 ) \
-    SYSCALL_ENTRY( 0x0009, NtAlertThreadByThreadId, 8 ) \
-    SYSCALL_ENTRY( 0x000a, NtAllocateLocallyUniqueId, 8 ) \
-    SYSCALL_ENTRY( 0x000b, NtAllocateReserveObject, 24 ) \
-    SYSCALL_ENTRY( 0x000c, NtAllocateUuids, 32 ) \
-    SYSCALL_ENTRY( 0x000d, NtAllocateVirtualMemory, 48 ) \
-    SYSCALL_ENTRY( 0x000e, NtAllocateVirtualMemoryEx, 56 ) \
-    SYSCALL_ENTRY( 0x000f, NtAreMappedFilesTheSame, 16 ) \
-    SYSCALL_ENTRY( 0x0010, NtAssignProcessToJobObject, 16 ) \
-    SYSCALL_ENTRY( 0x0011, NtCallbackReturn, 24 ) \
-    SYSCALL_ENTRY( 0x0012, NtCancelIoFile, 16 ) \
-    SYSCALL_ENTRY( 0x0013, NtCancelIoFileEx, 24 ) \
-    SYSCALL_ENTRY( 0x0014, NtCancelSynchronousIoFile, 24 ) \
-    SYSCALL_ENTRY( 0x0015, NtCancelTimer, 16 ) \
-    SYSCALL_ENTRY( 0x0016, NtClearEvent, 8 ) \
-    SYSCALL_ENTRY( 0x0017, NtClose, 8 ) \
-    SYSCALL_ENTRY( 0x0018, NtCommitTransaction, 16 ) \
-    SYSCALL_ENTRY( 0x0019, NtCompareObjects, 16 ) \
-    SYSCALL_ENTRY( 0x001a, NtCompareTokens, 24 ) \
-    SYSCALL_ENTRY( 0x001b, NtCompleteConnectPort, 8 ) \
-    SYSCALL_ENTRY( 0x001c, NtConnectPort, 64 ) \
-    SYSCALL_ENTRY( 0x001d, NtContinue, 16 ) \
-    SYSCALL_ENTRY( 0x001e, NtContinueEx, 16 ) \
-    SYSCALL_ENTRY( 0x001f, NtConvertBetweenAuxiliaryCounterAndPerformanceCounter, 32 ) \
-    SYSCALL_ENTRY( 0x0020, NtCreateDebugObject, 32 ) \
-    SYSCALL_ENTRY( 0x0021, NtCreateDirectoryObject, 24 ) \
-    SYSCALL_ENTRY( 0x0022, NtCreateEvent, 40 ) \
-    SYSCALL_ENTRY( 0x0023, NtCreateFile, 88 ) \
-    SYSCALL_ENTRY( 0x0024, NtCreateIoCompletion, 32 ) \
-    SYSCALL_ENTRY( 0x0025, NtCreateJobObject, 24 ) \
-    SYSCALL_ENTRY( 0x0026, NtCreateKey, 56 ) \
-    SYSCALL_ENTRY( 0x0027, NtCreateKeyTransacted, 64 ) \
-    SYSCALL_ENTRY( 0x0028, NtCreateKeyedEvent, 32 ) \
-    SYSCALL_ENTRY( 0x0029, NtCreateLowBoxToken, 72 ) \
-    SYSCALL_ENTRY( 0x002a, NtCreateMailslotFile, 64 ) \
-    SYSCALL_ENTRY( 0x002b, NtCreateMutant, 32 ) \
-    SYSCALL_ENTRY( 0x002c, NtCreateNamedPipeFile, 112 ) \
-    SYSCALL_ENTRY( 0x002d, NtCreatePagingFile, 32 ) \
-    SYSCALL_ENTRY( 0x002e, NtCreatePort, 40 ) \
-    SYSCALL_ENTRY( 0x002f, NtCreateSection, 56 ) \
-    SYSCALL_ENTRY( 0x0030, NtCreateSectionEx, 72 ) \
-    SYSCALL_ENTRY( 0x0031, NtCreateSemaphore, 40 ) \
-    SYSCALL_ENTRY( 0x0032, NtCreateSymbolicLinkObject, 32 ) \
-    SYSCALL_ENTRY( 0x0033, NtCreateThread, 64 ) \
-    SYSCALL_ENTRY( 0x0034, NtCreateThreadEx, 88 ) \
-    SYSCALL_ENTRY( 0x0035, NtCreateTimer, 32 ) \
-    SYSCALL_ENTRY( 0x0036, NtCreateToken, 104 ) \
-    SYSCALL_ENTRY( 0x0037, NtCreateTransaction, 80 ) \
-    SYSCALL_ENTRY( 0x0038, NtCreateUserProcess, 88 ) \
-    SYSCALL_ENTRY( 0x0039, NtDebugActiveProcess, 16 ) \
-    SYSCALL_ENTRY( 0x003a, NtDebugContinue, 24 ) \
-    SYSCALL_ENTRY( 0x003b, NtDelayExecution, 16 ) \
-    SYSCALL_ENTRY( 0x003c, NtDeleteAtom, 8 ) \
-    SYSCALL_ENTRY( 0x003d, NtDeleteFile, 8 ) \
-    SYSCALL_ENTRY( 0x003e, NtDeleteKey, 8 ) \
-    SYSCALL_ENTRY( 0x003f, NtDeleteValueKey, 16 ) \
-    SYSCALL_ENTRY( 0x0040, NtDeviceIoControlFile, 80 ) \
-    SYSCALL_ENTRY( 0x0041, NtDisplayString, 8 ) \
-    SYSCALL_ENTRY( 0x0042, NtDuplicateObject, 56 ) \
-    SYSCALL_ENTRY( 0x0043, NtDuplicateToken, 48 ) \
-    SYSCALL_ENTRY( 0x0044, NtEnumerateKey, 48 ) \
-    SYSCALL_ENTRY( 0x0045, NtEnumerateValueKey, 48 ) \
-    SYSCALL_ENTRY( 0x0046, NtFilterToken, 48 ) \
-    SYSCALL_ENTRY( 0x0047, NtFindAtom, 24 ) \
-    SYSCALL_ENTRY( 0x0048, NtFlushBuffersFile, 16 ) \
-    SYSCALL_ENTRY( 0x0049, NtFlushBuffersFileEx, 40 ) \
-    SYSCALL_ENTRY( 0x004a, NtFlushInstructionCache, 24 ) \
-    SYSCALL_ENTRY( 0x004b, NtFlushKey, 8 ) \
-    SYSCALL_ENTRY( 0x004c, NtFlushProcessWriteBuffers, 0 ) \
-    SYSCALL_ENTRY( 0x004d, NtFlushVirtualMemory, 32 ) \
-    SYSCALL_ENTRY( 0x004e, NtFreeVirtualMemory, 32 ) \
-    SYSCALL_ENTRY( 0x004f, NtFsControlFile, 80 ) \
-    SYSCALL_ENTRY( 0x0050, NtGetContextThread, 16 ) \
-    SYSCALL_ENTRY( 0x0051, NtGetCurrentProcessorNumber, 0 ) \
-    SYSCALL_ENTRY( 0x0052, NtGetNextProcess, 40 ) \
-    SYSCALL_ENTRY( 0x0053, NtGetNextThread, 48 ) \
-    SYSCALL_ENTRY( 0x0054, NtGetNlsSectionPtr, 40 ) \
-    SYSCALL_ENTRY( 0x0055, NtGetWriteWatch, 56 ) \
-    SYSCALL_ENTRY( 0x0056, NtImpersonateAnonymousToken, 8 ) \
-    SYSCALL_ENTRY( 0x0057, NtInitializeNlsFiles, 24 ) \
-    SYSCALL_ENTRY( 0x0058, NtInitiatePowerAction, 32 ) \
-    SYSCALL_ENTRY( 0x0059, NtIsProcessInJob, 16 ) \
-    SYSCALL_ENTRY( 0x005a, NtListenPort, 16 ) \
-    SYSCALL_ENTRY( 0x005b, NtLoadDriver, 8 ) \
-    SYSCALL_ENTRY( 0x005c, NtLoadKey, 16 ) \
-    SYSCALL_ENTRY( 0x005d, NtLoadKey2, 24 ) \
-    SYSCALL_ENTRY( 0x005e, NtLoadKeyEx, 64 ) \
-    SYSCALL_ENTRY( 0x005f, NtLockFile, 80 ) \
-    SYSCALL_ENTRY( 0x0060, NtLockVirtualMemory, 32 ) \
-    SYSCALL_ENTRY( 0x0061, NtMakePermanentObject, 8 ) \
-    SYSCALL_ENTRY( 0x0062, NtMakeTemporaryObject, 8 ) \
-    SYSCALL_ENTRY( 0x0063, NtMapViewOfSection, 80 ) \
-    SYSCALL_ENTRY( 0x0064, NtMapViewOfSectionEx, 72 ) \
-    SYSCALL_ENTRY( 0x0065, NtNotifyChangeDirectoryFile, 72 ) \
-    SYSCALL_ENTRY( 0x0066, NtNotifyChangeKey, 80 ) \
-    SYSCALL_ENTRY( 0x0067, NtNotifyChangeMultipleKeys, 96 ) \
-    SYSCALL_ENTRY( 0x0068, NtOpenDirectoryObject, 24 ) \
-    SYSCALL_ENTRY( 0x0069, NtOpenEvent, 24 ) \
-    SYSCALL_ENTRY( 0x006a, NtOpenFile, 48 ) \
-    SYSCALL_ENTRY( 0x006b, NtOpenIoCompletion, 24 ) \
-    SYSCALL_ENTRY( 0x006c, NtOpenJobObject, 24 ) \
-    SYSCALL_ENTRY( 0x006d, NtOpenKey, 24 ) \
-    SYSCALL_ENTRY( 0x006e, NtOpenKeyEx, 32 ) \
-    SYSCALL_ENTRY( 0x006f, NtOpenKeyTransacted, 32 ) \
-    SYSCALL_ENTRY( 0x0070, NtOpenKeyTransactedEx, 40 ) \
-    SYSCALL_ENTRY( 0x0071, NtOpenKeyedEvent, 24 ) \
-    SYSCALL_ENTRY( 0x0072, NtOpenMutant, 24 ) \
-    SYSCALL_ENTRY( 0x0073, NtOpenProcess, 32 ) \
-    SYSCALL_ENTRY( 0x0074, NtOpenProcessToken, 24 ) \
-    SYSCALL_ENTRY( 0x0075, NtOpenProcessTokenEx, 32 ) \
-    SYSCALL_ENTRY( 0x0076, NtOpenSection, 24 ) \
-    SYSCALL_ENTRY( 0x0077, NtOpenSemaphore, 24 ) \
-    SYSCALL_ENTRY( 0x0078, NtOpenSymbolicLinkObject, 24 ) \
-    SYSCALL_ENTRY( 0x0079, NtOpenThread, 32 ) \
-    SYSCALL_ENTRY( 0x007a, NtOpenThreadToken, 32 ) \
-    SYSCALL_ENTRY( 0x007b, NtOpenThreadTokenEx, 40 ) \
-    SYSCALL_ENTRY( 0x007c, NtOpenTimer, 24 ) \
-    SYSCALL_ENTRY( 0x007d, NtPowerInformation, 40 ) \
-    SYSCALL_ENTRY( 0x007e, NtPrivilegeCheck, 24 ) \
-    SYSCALL_ENTRY( 0x007f, NtProtectVirtualMemory, 40 ) \
-    SYSCALL_ENTRY( 0x0080, NtPulseEvent, 16 ) \
-    SYSCALL_ENTRY( 0x0081, NtQueryAttributesFile, 16 ) \
-    SYSCALL_ENTRY( 0x0082, NtQueryDefaultLocale, 16 ) \
-    SYSCALL_ENTRY( 0x0083, NtQueryDefaultUILanguage, 8 ) \
-    SYSCALL_ENTRY( 0x0084, NtQueryDirectoryFile, 88 ) \
-    SYSCALL_ENTRY( 0x0085, NtQueryDirectoryObject, 56 ) \
-    SYSCALL_ENTRY( 0x0086, NtQueryEaFile, 72 ) \
-    SYSCALL_ENTRY( 0x0087, NtQueryEvent, 40 ) \
-    SYSCALL_ENTRY( 0x0088, NtQueryFullAttributesFile, 16 ) \
-    SYSCALL_ENTRY( 0x0089, NtQueryInformationAtom, 40 ) \
-    SYSCALL_ENTRY( 0x008a, NtQueryInformationFile, 40 ) \
-    SYSCALL_ENTRY( 0x008b, NtQueryInformationJobObject, 40 ) \
-    SYSCALL_ENTRY_NtQueryInformationProcess( 0x008c, NtQueryInformationProcess, 40 ) \
-    SYSCALL_ENTRY( 0x008d, NtQueryInformationThread, 40 ) \
-    SYSCALL_ENTRY( 0x008e, NtQueryInformationToken, 40 ) \
-    SYSCALL_ENTRY( 0x008f, NtQueryInstallUILanguage, 8 ) \
-    SYSCALL_ENTRY( 0x0090, NtQueryIoCompletion, 40 ) \
-    SYSCALL_ENTRY( 0x0091, NtQueryKey, 40 ) \
-    SYSCALL_ENTRY( 0x0092, NtQueryLicenseValue, 40 ) \
-    SYSCALL_ENTRY( 0x0093, NtQueryMultipleValueKey, 48 ) \
-    SYSCALL_ENTRY( 0x0094, NtQueryMutant, 40 ) \
-    SYSCALL_ENTRY( 0x0095, NtQueryObject, 40 ) \
-    SYSCALL_ENTRY( 0x0096, NtQueryPerformanceCounter, 16 ) \
-    SYSCALL_ENTRY( 0x0097, NtQuerySection, 40 ) \
-    SYSCALL_ENTRY( 0x0098, NtQuerySecurityObject, 40 ) \
-    SYSCALL_ENTRY( 0x0099, NtQuerySemaphore, 40 ) \
-    SYSCALL_ENTRY( 0x009a, NtQuerySymbolicLinkObject, 24 ) \
-    SYSCALL_ENTRY( 0x009b, NtQuerySystemEnvironmentValue, 32 ) \
-    SYSCALL_ENTRY( 0x009c, NtQuerySystemEnvironmentValueEx, 40 ) \
-    SYSCALL_ENTRY( 0x009d, NtQuerySystemInformation, 32 ) \
-    SYSCALL_ENTRY( 0x009e, NtQuerySystemInformationEx, 48 ) \
-    SYSCALL_ENTRY_NtQuerySystemTime( 0x009f, NtQuerySystemTime, 8 ) \
-    SYSCALL_ENTRY( 0x00a0, NtQueryTimer, 40 ) \
-    SYSCALL_ENTRY( 0x00a1, NtQueryTimerResolution, 24 ) \
-    SYSCALL_ENTRY( 0x00a2, NtQueryValueKey, 48 ) \
-    SYSCALL_ENTRY( 0x00a3, NtQueryVirtualMemory, 48 ) \
-    SYSCALL_ENTRY( 0x00a4, NtQueryVolumeInformationFile, 40 ) \
-    SYSCALL_ENTRY( 0x00a5, NtQueueApcThread, 40 ) \
-    SYSCALL_ENTRY( 0x00a6, NtQueueApcThreadEx, 48 ) \
-    SYSCALL_ENTRY( 0x00a7, NtRaiseException, 24 ) \
-    SYSCALL_ENTRY( 0x00a8, NtRaiseHardError, 48 ) \
-    SYSCALL_ENTRY( 0x00a9, NtReadFile, 72 ) \
-    SYSCALL_ENTRY( 0x00aa, NtReadFileScatter, 72 ) \
-    SYSCALL_ENTRY( 0x00ab, NtReadVirtualMemory, 40 ) \
-    SYSCALL_ENTRY( 0x00ac, NtRegisterThreadTerminatePort, 8 ) \
-    SYSCALL_ENTRY( 0x00ad, NtReleaseKeyedEvent, 32 ) \
-    SYSCALL_ENTRY( 0x00ae, NtReleaseMutant, 16 ) \
-    SYSCALL_ENTRY( 0x00af, NtReleaseSemaphore, 24 ) \
-    SYSCALL_ENTRY( 0x00b0, NtRemoveIoCompletion, 40 ) \
-    SYSCALL_ENTRY( 0x00b1, NtRemoveIoCompletionEx, 48 ) \
-    SYSCALL_ENTRY( 0x00b2, NtRemoveProcessDebug, 16 ) \
-    SYSCALL_ENTRY( 0x00b3, NtRenameKey, 16 ) \
-    SYSCALL_ENTRY( 0x00b4, NtReplaceKey, 24 ) \
-    SYSCALL_ENTRY( 0x00b5, NtReplyWaitReceivePort, 32 ) \
-    SYSCALL_ENTRY( 0x00b6, NtRequestWaitReplyPort, 24 ) \
-    SYSCALL_ENTRY( 0x00b7, NtResetEvent, 16 ) \
-    SYSCALL_ENTRY( 0x00b8, NtResetWriteWatch, 24 ) \
-    SYSCALL_ENTRY( 0x00b9, NtRestoreKey, 24 ) \
-    SYSCALL_ENTRY( 0x00ba, NtResumeProcess, 8 ) \
-    SYSCALL_ENTRY( 0x00bb, NtResumeThread, 16 ) \
-    SYSCALL_ENTRY( 0x00bc, NtRollbackTransaction, 16 ) \
-    SYSCALL_ENTRY( 0x00bd, NtSaveKey, 16 ) \
-    SYSCALL_ENTRY( 0x00be, NtSecureConnectPort, 72 ) \
-    SYSCALL_ENTRY( 0x00bf, NtSetContextThread, 16 ) \
-    SYSCALL_ENTRY( 0x00c0, NtSetDebugFilterState, 24 ) \
-    SYSCALL_ENTRY( 0x00c1, NtSetDefaultLocale, 16 ) \
-    SYSCALL_ENTRY( 0x00c2, NtSetDefaultUILanguage, 8 ) \
-    SYSCALL_ENTRY( 0x00c3, NtSetEaFile, 32 ) \
-    SYSCALL_ENTRY( 0x00c4, NtSetEvent, 16 ) \
-    SYSCALL_ENTRY( 0x00c5, NtSetInformationDebugObject, 40 ) \
-    SYSCALL_ENTRY( 0x00c6, NtSetInformationFile, 40 ) \
-    SYSCALL_ENTRY( 0x00c7, NtSetInformationJobObject, 32 ) \
-    SYSCALL_ENTRY( 0x00c8, NtSetInformationKey, 32 ) \
-    SYSCALL_ENTRY( 0x00c9, NtSetInformationObject, 32 ) \
-    SYSCALL_ENTRY( 0x00ca, NtSetInformationProcess, 32 ) \
-    SYSCALL_ENTRY( 0x00cb, NtSetInformationThread, 32 ) \
-    SYSCALL_ENTRY( 0x00cc, NtSetInformationToken, 32 ) \
-    SYSCALL_ENTRY( 0x00cd, NtSetInformationVirtualMemory, 48 ) \
-    SYSCALL_ENTRY( 0x00ce, NtSetIntervalProfile, 16 ) \
-    SYSCALL_ENTRY( 0x00cf, NtSetIoCompletion, 40 ) \
-    SYSCALL_ENTRY( 0x00d0, NtSetIoCompletionEx, 48 ) \
-    SYSCALL_ENTRY( 0x00d1, NtSetLdtEntries, 32 ) \
-    SYSCALL_ENTRY( 0x00d2, NtSetSecurityObject, 24 ) \
-    SYSCALL_ENTRY( 0x00d3, NtSetSystemInformation, 24 ) \
-    SYSCALL_ENTRY( 0x00d4, NtSetSystemTime, 16 ) \
-    SYSCALL_ENTRY( 0x00d5, NtSetThreadExecutionState, 16 ) \
-    SYSCALL_ENTRY( 0x00d6, NtSetTimer, 56 ) \
-    SYSCALL_ENTRY( 0x00d7, NtSetTimerResolution, 24 ) \
-    SYSCALL_ENTRY( 0x00d8, NtSetValueKey, 48 ) \
-    SYSCALL_ENTRY( 0x00d9, NtSetVolumeInformationFile, 40 ) \
-    SYSCALL_ENTRY( 0x00da, NtShutdownSystem, 8 ) \
-    SYSCALL_ENTRY( 0x00db, NtSignalAndWaitForSingleObject, 32 ) \
-    SYSCALL_ENTRY( 0x00dc, NtSuspendProcess, 8 ) \
-    SYSCALL_ENTRY( 0x00dd, NtSuspendThread, 16 ) \
-    SYSCALL_ENTRY( 0x00de, NtSystemDebugControl, 48 ) \
-    SYSCALL_ENTRY( 0x00df, NtTerminateJobObject, 16 ) \
-    SYSCALL_ENTRY( 0x00e0, NtTerminateProcess, 16 ) \
-    SYSCALL_ENTRY( 0x00e1, NtTerminateThread, 16 ) \
-    SYSCALL_ENTRY( 0x00e2, NtTestAlert, 0 ) \
-    SYSCALL_ENTRY( 0x00e3, NtTraceControl, 48 ) \
-    SYSCALL_ENTRY( 0x00e4, NtUnloadDriver, 8 ) \
-    SYSCALL_ENTRY( 0x00e5, NtUnloadKey, 8 ) \
-    SYSCALL_ENTRY( 0x00e6, NtUnlockFile, 40 ) \
-    SYSCALL_ENTRY( 0x00e7, NtUnlockVirtualMemory, 32 ) \
-    SYSCALL_ENTRY( 0x00e8, NtUnmapViewOfSection, 16 ) \
-    SYSCALL_ENTRY( 0x00e9, NtUnmapViewOfSectionEx, 24 ) \
-    SYSCALL_ENTRY( 0x00ea, NtWaitForAlertByThreadId, 16 ) \
-    SYSCALL_ENTRY( 0x00eb, NtWaitForDebugEvent, 32 ) \
-    SYSCALL_ENTRY( 0x00ec, NtWaitForKeyedEvent, 32 ) \
-    SYSCALL_ENTRY( 0x00ed, NtWaitForMultipleObjects, 40 ) \
-    SYSCALL_ENTRY( 0x00ee, NtWaitForSingleObject, 24 ) \
-    SYSCALL_ENTRY( 0x00ef, NtWriteFile, 72 ) \
-    SYSCALL_ENTRY( 0x00f0, NtWriteFileGather, 72 ) \
-    SYSCALL_ENTRY( 0x00f1, NtWriteVirtualMemory, 40 ) \
-    SYSCALL_ENTRY( 0x00f2, NtYieldExecution, 0 ) \
+    SYSCALL_ENTRY( 0x0000, NtAccessCheck, 64 ) \
+    SYSCALL_ENTRY( 0x0001, NtAdjustGroupsToken, 48 ) \
+    SYSCALL_ENTRY( 0x0002, NtAcceptConnectPort, 48 ) \
+    SYSCALL_ENTRY( 0x0003, NtAlertMultipleThreadByThreadId, 32 ) \
+    SYSCALL_ENTRY( 0x0004, NtWaitForSingleObject, 24 ) \
+    SYSCALL_ENTRY( 0x0005, NtCallbackReturn, 24 ) \
+    SYSCALL_ENTRY( 0x0006, NtReadFile, 72 ) \
+    SYSCALL_ENTRY( 0x0007, NtDeviceIoControlFile, 80 ) \
+    SYSCALL_ENTRY( 0x0008, NtWriteFile, 72 ) \
+    SYSCALL_ENTRY( 0x0009, NtRemoveIoCompletion, 40 ) \
+    SYSCALL_ENTRY( 0x000a, NtReleaseSemaphore, 24 ) \
+    SYSCALL_ENTRY( 0x000b, NtReplyWaitReceivePort, 32 ) \
+    SYSCALL_ENTRY( 0x000c, NtAlertResumeThread, 16 ) \
+    SYSCALL_ENTRY( 0x000d, NtSetInformationThread, 32 ) \
+    SYSCALL_ENTRY( 0x000e, NtSetEvent, 16 ) \
+    SYSCALL_ENTRY( 0x000f, NtClose, 8 ) \
+    SYSCALL_ENTRY( 0x0010, NtQueryObject, 40 ) \
+    SYSCALL_ENTRY( 0x0011, NtQueryInformationFile, 40 ) \
+    SYSCALL_ENTRY( 0x0012, NtOpenKey, 24 ) \
+    SYSCALL_ENTRY( 0x0013, NtEnumerateValueKey, 48 ) \
+    SYSCALL_ENTRY( 0x0014, NtFindAtom, 24 ) \
+    SYSCALL_ENTRY( 0x0015, NtQueryDefaultLocale, 16 ) \
+    SYSCALL_ENTRY( 0x0016, NtQueryKey, 40 ) \
+    SYSCALL_ENTRY( 0x0017, NtQueryValueKey, 48 ) \
+    SYSCALL_ENTRY( 0x0018, NtAllocateVirtualMemory, 48 ) \
+    SYSCALL_ENTRY_NtQueryInformationProcess( 0x0019, NtQueryInformationProcess, 40 ) \
+    SYSCALL_ENTRY( 0x001a, NtAlertThread, 8 ) \
+    SYSCALL_ENTRY( 0x001b, NtWriteFileGather, 72 ) \
+    SYSCALL_ENTRY( 0x001c, NtSetInformationProcess, 32 ) \
+    SYSCALL_ENTRY( 0x001d, NtCreateKey, 56 ) \
+    SYSCALL_ENTRY( 0x001e, NtFreeVirtualMemory, 32 ) \
+    SYSCALL_ENTRY( 0x001f, NtAlertThreadByThreadId, 8 ) \
+    SYSCALL_ENTRY( 0x0020, NtReleaseMutant, 16 ) \
+    SYSCALL_ENTRY( 0x0021, NtQueryInformationToken, 40 ) \
+    SYSCALL_ENTRY( 0x0022, NtRequestWaitReplyPort, 24 ) \
+    SYSCALL_ENTRY( 0x0023, NtQueryVirtualMemory, 48 ) \
+    SYSCALL_ENTRY( 0x0024, NtOpenThreadToken, 32 ) \
+    SYSCALL_ENTRY( 0x0025, NtQueryInformationThread, 40 ) \
+    SYSCALL_ENTRY( 0x0026, NtOpenProcess, 32 ) \
+    SYSCALL_ENTRY( 0x0027, NtSetInformationFile, 40 ) \
+    SYSCALL_ENTRY( 0x0028, NtMapViewOfSection, 80 ) \
+    SYSCALL_ENTRY( 0x0029, NtAccessCheckAndAuditAlarm, 88 ) \
+    SYSCALL_ENTRY( 0x002a, NtUnmapViewOfSection, 16 ) \
+    SYSCALL_ENTRY( 0x002b, NtAllocateLocallyUniqueId, 8 ) \
+    SYSCALL_ENTRY( 0x002c, NtTerminateProcess, 16 ) \
+    SYSCALL_ENTRY( 0x002d, NtAllocateReserveObject, 24 ) \
+    SYSCALL_ENTRY( 0x002e, NtReadFileScatter, 72 ) \
+    SYSCALL_ENTRY( 0x002f, NtOpenThreadTokenEx, 40 ) \
+    SYSCALL_ENTRY( 0x0030, NtOpenProcessTokenEx, 32 ) \
+    SYSCALL_ENTRY( 0x0031, NtQueryPerformanceCounter, 16 ) \
+    SYSCALL_ENTRY( 0x0032, NtEnumerateKey, 48 ) \
+    SYSCALL_ENTRY( 0x0033, NtOpenFile, 48 ) \
+    SYSCALL_ENTRY( 0x0034, NtDelayExecution, 16 ) \
+    SYSCALL_ENTRY( 0x0035, NtQueryDirectoryFile, 88 ) \
+    SYSCALL_ENTRY( 0x0036, NtQuerySystemInformation, 32 ) \
+    SYSCALL_ENTRY( 0x0037, NtOpenSection, 24 ) \
+    SYSCALL_ENTRY( 0x0038, NtQueryTimer, 40 ) \
+    SYSCALL_ENTRY( 0x0039, NtFsControlFile, 80 ) \
+    SYSCALL_ENTRY( 0x003a, NtWriteVirtualMemory, 40 ) \
+    SYSCALL_ENTRY( 0x003b, NtAllocateUuids, 32 ) \
+    SYSCALL_ENTRY( 0x003c, NtDuplicateObject, 56 ) \
+    SYSCALL_ENTRY( 0x003d, NtQueryAttributesFile, 16 ) \
+    SYSCALL_ENTRY( 0x003e, NtClearEvent, 8 ) \
+    SYSCALL_ENTRY( 0x003f, NtReadVirtualMemory, 40 ) \
+    SYSCALL_ENTRY( 0x0040, NtOpenEvent, 24 ) \
+    SYSCALL_ENTRY( 0x0041, NtAdjustPrivilegesToken, 48 ) \
+    SYSCALL_ENTRY( 0x0042, NtDuplicateToken, 48 ) \
+    SYSCALL_ENTRY( 0x0043, NtContinue, 16 ) \
+    SYSCALL_ENTRY( 0x0044, NtQueryDefaultUILanguage, 8 ) \
+    SYSCALL_ENTRY( 0x0045, NtQueueApcThread, 40 ) \
+    SYSCALL_ENTRY( 0x0046, NtYieldExecution, 0 ) \
+    SYSCALL_ENTRY( 0x0047, NtAddAtom, 24 ) \
+    SYSCALL_ENTRY( 0x0048, NtCreateEvent, 40 ) \
+    SYSCALL_ENTRY( 0x0049, NtQueryVolumeInformationFile, 40 ) \
+    SYSCALL_ENTRY( 0x004a, NtCreateSection, 56 ) \
+    SYSCALL_ENTRY( 0x004b, NtFlushBuffersFile, 16 ) \
+    SYSCALL_ENTRY( 0x004c, NtAllocateVirtualMemoryEx, 56 ) \
+    SYSCALL_ENTRY( 0x004d, NtAreMappedFilesTheSame, 16 ) \
+    SYSCALL_ENTRY( 0x004e, NtCreateThread, 64 ) \
+    SYSCALL_ENTRY( 0x004f, NtIsProcessInJob, 16 ) \
+    SYSCALL_ENTRY( 0x0050, NtProtectVirtualMemory, 40 ) \
+    SYSCALL_ENTRY( 0x0051, NtQuerySection, 40 ) \
+    SYSCALL_ENTRY( 0x0052, NtResumeThread, 16 ) \
+    SYSCALL_ENTRY( 0x0053, NtTerminateThread, 16 ) \
+    SYSCALL_ENTRY( 0x0054, NtAssignProcessToJobObject, 16 ) \
+    SYSCALL_ENTRY( 0x0055, NtCreateFile, 88 ) \
+    SYSCALL_ENTRY( 0x0056, NtQueryEvent, 40 ) \
+    SYSCALL_ENTRY( 0x0057, NtCancelIoFileEx, 24 ) \
+    SYSCALL_ENTRY( 0x0058, NtOpenDirectoryObject, 24 ) \
+    SYSCALL_ENTRY( 0x0059, NtCancelSynchronousIoFile, 24 ) \
+    SYSCALL_ENTRY_NtQuerySystemTime( 0x005a, NtQuerySystemTime, 8 ) \
+    SYSCALL_ENTRY( 0x005b, NtWaitForMultipleObjects, 40 ) \
+    SYSCALL_ENTRY( 0x005c, NtSetInformationObject, 32 ) \
+    SYSCALL_ENTRY( 0x005d, NtCancelIoFile, 16 ) \
+    SYSCALL_ENTRY( 0x005e, NtCommitTransaction, 16 ) \
+    SYSCALL_ENTRY( 0x005f, NtPowerInformation, 40 ) \
+    SYSCALL_ENTRY( 0x0060, NtSetValueKey, 48 ) \
+    SYSCALL_ENTRY( 0x0061, NtCancelTimer, 16 ) \
+    SYSCALL_ENTRY( 0x0062, NtSetTimer, 56 ) \
+    SYSCALL_ENTRY( 0x0063, NtCompareObjects, 16 ) \
+    SYSCALL_ENTRY( 0x0064, NtCompareTokens, 24 ) \
+    SYSCALL_ENTRY( 0x0065, NtCompleteConnectPort, 8 ) \
+    SYSCALL_ENTRY( 0x0066, NtConnectPort, 64 ) \
+    SYSCALL_ENTRY( 0x0067, NtContinueEx, 16 ) \
+    SYSCALL_ENTRY( 0x0068, NtConvertBetweenAuxiliaryCounterAndPerformanceCounter, 32 ) \
+    SYSCALL_ENTRY( 0x0069, NtCreateDirectoryObject, 24 ) \
+    SYSCALL_ENTRY( 0x006a, NtCreateIoCompletion, 32 ) \
+    SYSCALL_ENTRY( 0x006b, NtCreateJobObject, 24 ) \
+    SYSCALL_ENTRY( 0x006c, NtCreateKeyTransacted, 64 ) \
+    SYSCALL_ENTRY( 0x006d, NtCreateKeyedEvent, 32 ) \
+    SYSCALL_ENTRY( 0x006e, NtCreateLowBoxToken, 72 ) \
+    SYSCALL_ENTRY( 0x006f, NtCreateMailslotFile, 64 ) \
+    SYSCALL_ENTRY( 0x0070, NtCreateMutant, 32 ) \
+    SYSCALL_ENTRY( 0x0071, NtCreateNamedPipeFile, 112 ) \
+    SYSCALL_ENTRY( 0x0072, NtCreatePagingFile, 32 ) \
+    SYSCALL_ENTRY( 0x0073, NtCreatePort, 40 ) \
+    SYSCALL_ENTRY( 0x0074, NtCreateSectionEx, 72 ) \
+    SYSCALL_ENTRY( 0x0075, NtCreateSemaphore, 40 ) \
+    SYSCALL_ENTRY( 0x0076, NtCreateSymbolicLinkObject, 32 ) \
+    SYSCALL_ENTRY( 0x0077, NtCreateThreadEx, 88 ) \
+    SYSCALL_ENTRY( 0x0078, NtCreateTimer, 32 ) \
+    SYSCALL_ENTRY( 0x0079, NtCreateToken, 104 ) \
+    SYSCALL_ENTRY( 0x007a, NtCreateTransaction, 80 ) \
+    SYSCALL_ENTRY( 0x007b, NtCreateUserProcess, 88 ) \
+    SYSCALL_ENTRY( 0x007c, NtDebugActiveProcess, 16 ) \
+    SYSCALL_ENTRY( 0x007d, NtDebugContinue, 24 ) \
+    SYSCALL_ENTRY( 0x007e, NtDeleteAtom, 8 ) \
+    SYSCALL_ENTRY( 0x007f, NtDeleteFile, 8 ) \
+    SYSCALL_ENTRY( 0x0080, NtDeleteKey, 8 ) \
+    SYSCALL_ENTRY( 0x0081, NtDeleteValueKey, 16 ) \
+    SYSCALL_ENTRY( 0x0082, NtDisplayString, 8 ) \
+    SYSCALL_ENTRY( 0x0083, NtFilterToken, 48 ) \
+    SYSCALL_ENTRY( 0x0084, NtFlushBuffersFileEx, 40 ) \
+    SYSCALL_ENTRY( 0x0085, NtFlushInstructionCache, 24 ) \
+    SYSCALL_ENTRY( 0x0086, NtFlushKey, 8 ) \
+    SYSCALL_ENTRY( 0x0087, NtFlushProcessWriteBuffers, 0 ) \
+    SYSCALL_ENTRY( 0x0088, NtFlushVirtualMemory, 32 ) \
+    SYSCALL_ENTRY( 0x0089, NtGetContextThread, 16 ) \
+    SYSCALL_ENTRY( 0x008a, NtGetCurrentProcessorNumber, 0 ) \
+    SYSCALL_ENTRY( 0x008b, NtGetNextProcess, 40 ) \
+    SYSCALL_ENTRY( 0x008c, NtGetNextThread, 48 ) \
+    SYSCALL_ENTRY( 0x008d, NtGetNlsSectionPtr, 40 ) \
+    SYSCALL_ENTRY( 0x008e, NtGetWriteWatch, 56 ) \
+    SYSCALL_ENTRY( 0x008f, NtImpersonateAnonymousToken, 8 ) \
+    SYSCALL_ENTRY( 0x0090, NtInitializeNlsFiles, 24 ) \
+    SYSCALL_ENTRY( 0x0091, NtInitiatePowerAction, 32 ) \
+    SYSCALL_ENTRY( 0x0092, NtListenPort, 16 ) \
+    SYSCALL_ENTRY( 0x0093, NtLoadDriver, 8 ) \
+    SYSCALL_ENTRY( 0x0094, NtLoadKey, 16 ) \
+    SYSCALL_ENTRY( 0x0095, NtLoadKey2, 24 ) \
+    SYSCALL_ENTRY( 0x0096, NtLoadKeyEx, 64 ) \
+    SYSCALL_ENTRY( 0x0097, NtLockFile, 80 ) \
+    SYSCALL_ENTRY( 0x0098, NtLockVirtualMemory, 32 ) \
+    SYSCALL_ENTRY( 0x0099, NtMakePermanentObject, 8 ) \
+    SYSCALL_ENTRY( 0x009a, NtMakeTemporaryObject, 8 ) \
+    SYSCALL_ENTRY( 0x009b, NtMapViewOfSectionEx, 72 ) \
+    SYSCALL_ENTRY( 0x009c, NtNotifyChangeDirectoryFile, 72 ) \
+    SYSCALL_ENTRY( 0x009d, NtNotifyChangeKey, 80 ) \
+    SYSCALL_ENTRY( 0x009e, NtNotifyChangeMultipleKeys, 96 ) \
+    SYSCALL_ENTRY( 0x009f, NtOpenIoCompletion, 24 ) \
+    SYSCALL_ENTRY( 0x00a0, NtOpenJobObject, 24 ) \
+    SYSCALL_ENTRY( 0x00a1, NtOpenKeyEx, 32 ) \
+    SYSCALL_ENTRY( 0x00a2, NtOpenKeyTransacted, 32 ) \
+    SYSCALL_ENTRY( 0x00a3, NtOpenKeyTransactedEx, 40 ) \
+    SYSCALL_ENTRY( 0x00a4, NtOpenKeyedEvent, 24 ) \
+    SYSCALL_ENTRY( 0x00a5, NtOpenMutant, 24 ) \
+    SYSCALL_ENTRY( 0x00a6, NtCreateDebugObject, 32 ) \
+    SYSCALL_ENTRY( 0x00a7, NtOpenProcessToken, 24 ) \
+    SYSCALL_ENTRY( 0x00a8, NtOpenSemaphore, 24 ) \
+    SYSCALL_ENTRY( 0x00a9, NtOpenSymbolicLinkObject, 24 ) \
+    SYSCALL_ENTRY( 0x00aa, NtOpenThread, 32 ) \
+    SYSCALL_ENTRY( 0x00ab, NtOpenTimer, 24 ) \
+    SYSCALL_ENTRY( 0x00ac, NtPrivilegeCheck, 24 ) \
+    SYSCALL_ENTRY( 0x00ad, NtPulseEvent, 16 ) \
+    SYSCALL_ENTRY( 0x00ae, NtQueryDirectoryObject, 56 ) \
+    SYSCALL_ENTRY( 0x00af, NtQueryEaFile, 72 ) \
+    SYSCALL_ENTRY( 0x00b0, NtQueryFullAttributesFile, 16 ) \
+    SYSCALL_ENTRY( 0x00b1, NtQueryInformationAtom, 40 ) \
+    SYSCALL_ENTRY( 0x00b2, NtQueryInformationJobObject, 40 ) \
+    SYSCALL_ENTRY( 0x00b3, NtQueryInstallUILanguage, 8 ) \
+    SYSCALL_ENTRY( 0x00b4, NtQueryIoCompletion, 40 ) \
+    SYSCALL_ENTRY( 0x00b5, NtQueryLicenseValue, 40 ) \
+    SYSCALL_ENTRY( 0x00b6, NtQueryMultipleValueKey, 48 ) \
+    SYSCALL_ENTRY( 0x00b7, NtQueryMutant, 40 ) \
+    SYSCALL_ENTRY( 0x00b8, NtQuerySecurityObject, 40 ) \
+    SYSCALL_ENTRY( 0x00b9, NtQuerySemaphore, 40 ) \
+    SYSCALL_ENTRY( 0x00ba, NtQuerySymbolicLinkObject, 24 ) \
+    SYSCALL_ENTRY( 0x00bb, NtQuerySystemEnvironmentValue, 32 ) \
+    SYSCALL_ENTRY( 0x00bc, NtQuerySystemEnvironmentValueEx, 40 ) \
+    SYSCALL_ENTRY( 0x00bd, NtQuerySystemInformationEx, 48 ) \
+    SYSCALL_ENTRY( 0x00be, NtQueryTimerResolution, 24 ) \
+    SYSCALL_ENTRY( 0x00bf, NtQueueApcThreadEx, 48 ) \
+    SYSCALL_ENTRY( 0x00c0, NtRaiseException, 24 ) \
+    SYSCALL_ENTRY( 0x00c1, NtRaiseHardError, 48 ) \
+    SYSCALL_ENTRY( 0x00c2, NtRegisterThreadTerminatePort, 8 ) \
+    SYSCALL_ENTRY( 0x00c3, NtReleaseKeyedEvent, 32 ) \
+    SYSCALL_ENTRY( 0x00c4, NtRemoveIoCompletionEx, 48 ) \
+    SYSCALL_ENTRY( 0x00c5, NtRemoveProcessDebug, 16 ) \
+    SYSCALL_ENTRY( 0x00c6, NtRenameKey, 16 ) \
+    SYSCALL_ENTRY( 0x00c7, NtReplaceKey, 24 ) \
+    SYSCALL_ENTRY( 0x00c8, NtResetEvent, 16 ) \
+    SYSCALL_ENTRY( 0x00c9, NtResetWriteWatch, 24 ) \
+    SYSCALL_ENTRY( 0x00ca, NtRestoreKey, 24 ) \
+    SYSCALL_ENTRY( 0x00cb, NtResumeProcess, 8 ) \
+    SYSCALL_ENTRY( 0x00cc, NtRollbackTransaction, 16 ) \
+    SYSCALL_ENTRY( 0x00cd, NtSaveKey, 16 ) \
+    SYSCALL_ENTRY( 0x00ce, NtSecureConnectPort, 72 ) \
+    SYSCALL_ENTRY( 0x00cf, NtSetContextThread, 16 ) \
+    SYSCALL_ENTRY( 0x00d0, NtSetDebugFilterState, 24 ) \
+    SYSCALL_ENTRY( 0x00d1, NtSetDefaultLocale, 16 ) \
+    SYSCALL_ENTRY( 0x00d2, NtSetDefaultUILanguage, 8 ) \
+    SYSCALL_ENTRY( 0x00d3, NtSetEaFile, 32 ) \
+    SYSCALL_ENTRY( 0x00d4, NtSetInformationDebugObject, 40 ) \
+    SYSCALL_ENTRY( 0x00d5, NtSetInformationJobObject, 32 ) \
+    SYSCALL_ENTRY( 0x00d6, NtSetInformationKey, 32 ) \
+    SYSCALL_ENTRY( 0x00d7, NtSetInformationToken, 32 ) \
+    SYSCALL_ENTRY( 0x00d8, NtSetInformationVirtualMemory, 48 ) \
+    SYSCALL_ENTRY( 0x00d9, NtSetIntervalProfile, 16 ) \
+    SYSCALL_ENTRY( 0x00da, NtSetIoCompletion, 40 ) \
+    SYSCALL_ENTRY( 0x00db, NtSetIoCompletionEx, 48 ) \
+    SYSCALL_ENTRY( 0x00dc, NtSetLdtEntries, 32 ) \
+    SYSCALL_ENTRY( 0x00dd, NtSetSecurityObject, 24 ) \
+    SYSCALL_ENTRY( 0x00de, NtSetSystemInformation, 24 ) \
+    SYSCALL_ENTRY( 0x00df, NtSetSystemTime, 16 ) \
+    SYSCALL_ENTRY( 0x00e0, NtSetThreadExecutionState, 16 ) \
+    SYSCALL_ENTRY( 0x00e1, NtSetTimerResolution, 24 ) \
+    SYSCALL_ENTRY( 0x00e2, NtSetVolumeInformationFile, 40 ) \
+    SYSCALL_ENTRY( 0x00e3, NtShutdownSystem, 8 ) \
+    SYSCALL_ENTRY( 0x00e4, NtSignalAndWaitForSingleObject, 32 ) \
+    SYSCALL_ENTRY( 0x00e5, NtSuspendProcess, 8 ) \
+    SYSCALL_ENTRY( 0x00e6, NtSuspendThread, 16 ) \
+    SYSCALL_ENTRY( 0x00e7, NtSystemDebugControl, 48 ) \
+    SYSCALL_ENTRY( 0x00e8, NtTerminateJobObject, 16 ) \
+    SYSCALL_ENTRY( 0x00e9, NtTestAlert, 0 ) \
+    SYSCALL_ENTRY( 0x00ea, NtTraceControl, 48 ) \
+    SYSCALL_ENTRY( 0x00eb, NtUnloadDriver, 8 ) \
+    SYSCALL_ENTRY( 0x00ec, NtUnloadKey, 8 ) \
+    SYSCALL_ENTRY( 0x00ed, NtUnlockFile, 40 ) \
+    SYSCALL_ENTRY( 0x00ee, NtUnlockVirtualMemory, 32 ) \
+    SYSCALL_ENTRY( 0x00ef, NtUnmapViewOfSectionEx, 24 ) \
+    SYSCALL_ENTRY( 0x00f0, NtWaitForAlertByThreadId, 16 ) \
+    SYSCALL_ENTRY( 0x00f1, NtWaitForDebugEvent, 32 ) \
+    SYSCALL_ENTRY( 0x00f2, NtWaitForKeyedEvent, 32 ) \
     SYSCALL_ENTRY( 0x00f3, wine_nt_to_unix_file_name, 32 ) \
     SYSCALL_ENTRY( 0x00f4, wine_unix_to_nt_file_name, 24 )
 #else
-- 
2.51.1

